<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="DEP2">
<title>Data import • DEP2</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data import">
<meta property="og:description" content="DEP2">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DEP2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.05.23</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Data_import.html">Data import</a>
    <a class="dropdown-item" href="../articles/Quick_start.html">Quick start</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Data_import_files/htmlwidgets-1.6.1/htmlwidgets.js"></script><link href="Data_import_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="Data_import_files/datatables-binding-0.27/datatables.js"></script><link href="Data_import_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="Data_import_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="Data_import_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="Data_import_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="Data_import_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Data import</h1>
                        <h4 data-toc-skip class="author">Zhenhuan
Feng</h4>
                        
      
      
      <div class="d-none name"><code>Data_import.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vigenette introduce the how DEP2 receives upstream
      quantitative results from MaxQuant, DIA-NN, Spectronaut, or
      Fragpipe.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>DEP2</code> use the <code>make_unique</code> function to format
protein/PTM-peptide-level table and load them into S4 objects via
<code>make_se/pe</code> function. The parameters in these functions are
adjustable to suit different result tables out from search and
quantitative software. Here we demonstrate the workflow and detail on
inputs from four quantitative software.</p>
</div>
<div class="section level2">
<h2 id="example-data-acquirement">Example data acquirement<a class="anchor" aria-label="anchor" href="#example-data-acquirement"></a>
</h2>
<p>Here we use a benchmark proteomics data by spiking Yeast and E.coli
lysate into Hela background. Raw data is acquired through an Orbitrap
Fusion Lumos Tribrid equipment in either DDA or DIA mode. The DDA data
is search and quantified by MaxQuant or FragPipe respectively . And the
DIA data is analysed search by DIA-NN or Spectronaut (library-free
mode). The example tables is available in this <a href="github.com/mildpiggy/OmicsExample">repository</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download OmicsExample and uncompress firstly</span></span>
<span></span>
<span><span class="co"># Change the path to Omics Example</span></span>
<span><span class="va">example_path</span> <span class="op">=</span> <span class="st">"the/path/to/OmicsExample"</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span></span>
<span>  root.dir <span class="op">=</span> <span class="va">example_path</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEP2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="construct-s4-object">Construct S4 object<a class="anchor" aria-label="anchor" href="#construct-s4-object"></a>
</h2>
<p>The DEP2 analysis workflow is build upon S4 object. Therefore the
first step is to format table and convert it into S4 container.</p>
<div class="section level3">
<h3 id="convert-se-from-protein-level-data">Convert SE from protein-level data<a class="anchor" aria-label="anchor" href="#convert-se-from-protein-level-data"></a>
</h3>
<p>For protein-level data, use gene/protein as feature identifier. The
entire workflow around the <em>SummarizedExperiment</em> object. For
different input data, DEP2 provides <code>make_unique</code> and
<code>make_unique_ptm</code> to format feature identifier in proteomics
result. <code>make_unqiue</code> function clean identifiers, keep the
first name/id in each group, and make repeated identifiers unique by add
suffix (‘ProXXX’, ‘ProXXX.1’, ‘ProXXX.2’). Repeated identifiers are
common when library search is performed with isoforms or unreviewed
sequences. Parameter <em>names</em> and <em>ids</em> should be
gene/protein names or ids. And in the output, unique identifiers are
saved in the new columns “name” and “id”.</p>
<div class="section level4">
<h4 id="from-maxquant-proteingroups">From MaxQuant proteinGroups<a class="anchor" aria-label="anchor" href="#from-maxquant-proteingroups"></a>
</h4>
<p>In this pipeline, DEP2 use the table <code>proteinGroups.txt</code>
as input.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load proteinGroups table</span></span>
<span><span class="va">mq_pgfile</span> <span class="op">&lt;-</span> <span class="st">"./A_spikeIn/MaxQuant_output/proteinGroups.txt.gz"</span></span>
<span></span>
<span><span class="va">pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">mq_pgfile</span>,sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div>
<p>The <em>names</em> and <em>ids</em> are in the columns “Gene.names”
and “Gene.names”.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate unique names and ids</span></span>
<span><span class="va">unique_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">pg</span>, </span>
<span>                         name <span class="op">=</span> <span class="st">"Gene.names"</span>, <span class="co">#gene </span></span>
<span>                         ids <span class="op">=</span> <span class="st">"Protein.IDs"</span>   <span class="co">#protein</span></span>
<span>                         <span class="op">)</span></span>
<span></span>
<span><span class="co">## new columns name and id, which is necessary to make_se</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">unique_pg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"name"</span>,<span class="st">"ID"</span>,<span class="st">"Gene.names"</span>,<span class="st">"Protein.IDs"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;         name         ID</span></span>
<span><span class="co">#&gt; 1      NUDT4     Q9NZJ9</span></span>
<span><span class="co">#&gt; 2     P0DPI2     P0DPI2</span></span>
<span><span class="co">#&gt; 3 A0A0U1RRE5 A0A0U1RRE5</span></span>
<span><span class="co">#&gt; 4 A0A0U1RRL7 A0A0U1RRL7</span></span>
<span><span class="co">#&gt; 5     JW5951 A0A385XJE6</span></span>
<span><span class="co">#&gt; 6       UBA6     A0AVT1</span></span>
<span><span class="co">#&gt;                                                                                                             Gene.names</span></span>
<span><span class="co">#&gt; 1                                                                                                                NUDT4</span></span>
<span><span class="co">#&gt; 2                                                                                                                     </span></span>
<span><span class="co">#&gt; 3                                                                                                                     </span></span>
<span><span class="co">#&gt; 4                                                                                                                     </span></span>
<span><span class="co">#&gt; 5 JW5951;JW5936;JW5935;JW5928;JW5914;JW5895;JW1882;insH11;insH10;insH9;insH8;insH7;insH6;insH4;insH3;insH2;insH1;insH5</span></span>
<span><span class="co">#&gt; 6                                                                                                                 UBA6</span></span>
<span><span class="co">#&gt;                                                                                                                                Protein.IDs</span></span>
<span><span class="co">#&gt; 1                                                                                                          Q9NZJ9;A0A024RBG1;Q96G61;Q8NFP7</span></span>
<span><span class="co">#&gt; 2                                                                                                                        P0DPI2;A0A0B4J2D5</span></span>
<span><span class="co">#&gt; 3                                                                                                                               A0A0U1RRE5</span></span>
<span><span class="co">#&gt; 4                                                                                                                               A0A0U1RRL7</span></span>
<span><span class="co">#&gt; 5 A0A385XJE6;P0CE65;P0CE64;P0CE63;P0CE62;P0CE61;P0CE60;P0CE59;P0CE58;P0CE57;P0CE56;P0CE55;P0CE54;P0CE53;P0CE52;P0CE51;P0CE50;P0CE49;P76071</span></span>
<span><span class="co">#&gt; 6                                                                                                                                   A0AVT1</span></span></code></pre></div>
<p>In this pipeline DEP2 uses SE object as the container to store
expression assay, features information (row data) and experiment design
(columns data). After <code>make_unique</code> identifiers, DEP2
provides <code>make_se</code>/<code>make_se_parse</code> to convert
<code>SummarizedExperiment</code> object. The former required an input
experiment design table to annotate samples, which must contain three
columns ‘label’, ‘condition’ and ‘replicate’. While the ‘parse’ function
could automatically assign experiment design by parsing abundance assay
column names.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Expression col in the input data</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"LFQ.intensity."</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_pg</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Experiment design table</span></span>
<span><span class="va">expDesign_file</span> <span class="op">&lt;-</span> <span class="st">"./A_spikeIn/MaxQuant_output/expdesign.txt.gz"</span></span>
<span><span class="va">expdesign</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">expDesign_file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">expdesign</span></span>
<span><span class="co">#&gt;   label condition replicate</span></span>
<span><span class="co">#&gt; 1  A1_1        A1         1</span></span>
<span><span class="co">#&gt; 2  A1_2        A1         2</span></span>
<span><span class="co">#&gt; 3  A1_3        A1         3</span></span>
<span><span class="co">#&gt; 4  B1_1        B1         1</span></span>
<span><span class="co">#&gt; 5  B1_2        B1         2</span></span>
<span><span class="co">#&gt; 6  B1_3        B1         3</span></span>
<span></span>
<span><span class="co">## Convert se with expdesign table</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se.html">make_se</a></span><span class="op">(</span><span class="va">unique_pg</span>, columns <span class="op">=</span> <span class="va">ecols</span>, expdesign <span class="op">=</span> <span class="va">expdesign</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Convert se with expdesign table</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se_parse.html">make_se_parse</a></span><span class="op">(</span><span class="va">unique_pg</span>, columns <span class="op">=</span> <span class="va">ecols</span>, mode <span class="op">=</span> <span class="st">"delim"</span><span class="op">)</span></span></code></pre></div>
<p>How experiment design is extracted from colnames through</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## How experiment design is extracted from colnames. Split by delim or by character number </span></span>
<span><span class="va">sample_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quantity.A_1"</span>,<span class="st">"Quantity.A_2"</span>,<span class="st">"Quantity.B_1"</span>,</span>
<span>                 <span class="st">"Quantity.B_3"</span>,<span class="st">"Quantity.B_2"</span>,<span class="st">"Quantity.A_3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_exdesign_parse.html">get_exdesign_parse</a></span><span class="op">(</span><span class="va">sample_names</span>,mode <span class="op">=</span> <span class="st">"delim"</span>, sep <span class="op">=</span> <span class="st">"_"</span>, remove_prefix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   label  ID condition replicate</span></span>
<span><span class="co">#&gt; 1   A_1 A_1         A         1</span></span>
<span><span class="co">#&gt; 2   A_2 A_2         A         2</span></span>
<span><span class="co">#&gt; 3   B_1 B_1         B         1</span></span>
<span><span class="co">#&gt; 4   B_3 B_3         B         3</span></span>
<span><span class="co">#&gt; 5   B_2 B_2         B         2</span></span>
<span><span class="co">#&gt; 6   A_3 A_3         A         3</span></span>
<span></span>
<span><span class="va">sample_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quantity.A1"</span>,<span class="st">"Quantity.A2"</span>,<span class="st">"Quantity.B1"</span>,</span>
<span>                 <span class="st">"Quantity.B3"</span>,<span class="st">"Quantity.B2"</span>,<span class="st">"Quantity.A3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_exdesign_parse.html">get_exdesign_parse</a></span><span class="op">(</span><span class="va">sample_names</span>,mode <span class="op">=</span> <span class="st">"char"</span>, chars <span class="op">=</span> <span class="fl">1</span>, remove_prefix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   label ID condition replicate</span></span>
<span><span class="co">#&gt; 1    A1 A1         A         1</span></span>
<span><span class="co">#&gt; 2    A2 A2         A         2</span></span>
<span><span class="co">#&gt; 3    B1 B1         B         1</span></span>
<span><span class="co">#&gt; 4    B3 B3         B         3</span></span>
<span><span class="co">#&gt; 5    B2 B2         B         2</span></span>
<span><span class="co">#&gt; 6    A3 A3         A         3</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="from-fragpipe-combined_protein">From FragPipe combined_protein<a class="anchor" aria-label="anchor" href="#from-fragpipe-combined_protein"></a>
</h4>
<p>In this pipeline, DEP2 use the table
<code>combined_protein.tsv</code> from FragPipe as input.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load combined_protein.tsv table</span></span>
<span><span class="va">FragP_pgfile</span> <span class="op">&lt;-</span> <span class="st">"./A_spikeIn/FragPipe_output/combined_protein.tsv.gz"</span></span>
<span></span>
<span><span class="va">FragP_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">FragP_pgfile</span>,sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Unique identifier</strong>. The <em>names</em> and
<em>ids</em> are in the columns “Gene” and “Protein.ID”.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate unique names and ids</span></span>
<span><span class="va">unique_FragP_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">FragP_pg</span>, </span>
<span>                               name <span class="op">=</span> <span class="st">"Gene"</span>,  <span class="co">#gene </span></span>
<span>                               ids <span class="op">=</span> <span class="st">"Protein.ID"</span>   <span class="co">#protein</span></span>
<span>                               <span class="op">)</span></span>
<span></span>
<span><span class="co">## new columns name and id, which is necessary to make_se</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">unique_FragP_pg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"ID"</span>, <span class="st">"Gene"</span>, <span class="st">"Protein.ID"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;     name     ID   Gene Protein.ID</span></span>
<span><span class="co">#&gt; 1   ADH1 P00330   ADH1     P00330</span></span>
<span><span class="co">#&gt; 2 P00761 P00761            P00761</span></span>
<span><span class="co">#&gt; 3 CSN1S1 P02662 CSN1S1     P02662</span></span>
<span><span class="co">#&gt; 4 CSN1S2 P02663 CSN1S2     P02663</span></span>
<span><span class="co">#&gt; 5   CSN2 P02666   CSN2     P02666</span></span>
<span><span class="co">#&gt; 6   CSN3 P02668   CSN3     P02668</span></span></code></pre></div>
<p><strong>Convert SE</strong>. Here, We used the “MaxLFQ Intensity”
values. “Intesity” is also practicable, but may bring some effect on
test result.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Expression col in the combined_protein.tsv</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">".MaxLFQ.Intensity"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_FragP_pg</span><span class="op">)</span>,value <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">ecols</span></span>
<span><span class="co">#&gt; [1] "A_1.MaxLFQ.Intensity" "A_2.MaxLFQ.Intensity" "A_3.MaxLFQ.Intensity"</span></span>
<span><span class="co">#&gt; [4] "B_1.MaxLFQ.Intensity" "B_3.MaxLFQ.Intensity"</span></span>
<span></span>
<span><span class="co">## Convert SE with expdesign table</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se_parse.html">make_se_parse</a></span><span class="op">(</span><span class="va">unique_FragP_pg</span>, </span>
<span>                    columns <span class="op">=</span> <span class="va">ecols</span>, </span>
<span>                    mode <span class="op">=</span> <span class="st">"delim"</span>,</span>
<span>                    remove_suffix <span class="op">=</span> <span class="cn">TRUE</span>    <span class="co">## remove the col suffix ".MaxLFQ.Intensity"</span></span>
<span>                    <span class="op">)</span></span>
<span></span>
<span><span class="co">## The suffix is removed in SE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 5 rows and 4 columns</span></span>
<span><span class="co">#&gt;           label          ID   condition   replicate</span></span>
<span><span class="co">#&gt;     &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; A_1         A_1         A_1           A           1</span></span>
<span><span class="co">#&gt; A_2         A_2         A_2           A           2</span></span>
<span><span class="co">#&gt; A_3         A_3         A_3           A           3</span></span>
<span><span class="co">#&gt; B_1         B_1         B_1           B           1</span></span>
<span><span class="co">#&gt; B_3         B_3         B_3           B           3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A_1" "A_2" "A_3" "B_1" "B_3"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="from-dia-nn-result">From DIA-NN result<a class="anchor" aria-label="anchor" href="#from-dia-nn-result"></a>
</h4>
<p>As in the example above, <code>make_se</code> and
<code>make_se_parse</code> accept wide-format table which each row means
a features (a protein or a proteingroup), and rows contain the
identification information and quantification information of samples.
However, the long-format tables are also widely used in many cases. For
example the <em>report.tsv</em> output from DIA-NN or the
<em>MSstats</em> output. Function <code>reshape_long2wide</code> can
turn long-format into wide table. Next, we will show the difference
between handling wide- or long-table.</p>
<div class="section level5">
<h5 id="wide-table-output">Wide table output<a class="anchor" aria-label="anchor" href="#wide-table-output"></a>
</h5>
<p>The <em>report.pg_matrix.tsv</em> from DIA-NN is a wide-format table
similar to proteinGroups.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load combined_protein.tsv table</span></span>
<span><span class="va">Diann_pgfile</span> <span class="op">&lt;-</span> <span class="st">"./A_spikeIn/Diann_output/report.pg_matrix.tsv.gz"</span></span>
<span></span>
<span><span class="va">Diann_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">Diann_pgfile</span>,sep <span class="op">=</span> <span class="st">"\t"</span>, fileEncoding<span class="op">=</span><span class="st">"latin1"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Unique identifier</strong>. The <em>names</em> and
<em>Protein.Group</em> are in the columns “Genes” and
“Protein.Group”.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate unique names and ids</span></span>
<span><span class="va">unique_diann_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">Diann_pg</span>, </span>
<span>                               name <span class="op">=</span> <span class="st">"Genes"</span>,  <span class="co">#gene </span></span>
<span>                               ids <span class="op">=</span> <span class="st">"Protein.Group"</span>   <span class="co">#protein</span></span>
<span>                               <span class="op">)</span></span>
<span></span>
<span><span class="co">## new columns name and id, which is necessary to make_se</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">unique_diann_pg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"ID"</span>, <span class="st">"Genes"</span>, <span class="st">"Protein.Group"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;        name         ID           Genes     Protein.Group</span></span>
<span><span class="co">#&gt; 1 YER079C-A A0A023PZB8       YER079C-A        A0A023PZB8</span></span>
<span><span class="co">#&gt; 2   PPIAL4C A0A0B4J2A2 PPIAL4C;PPIAL4G A0A0B4J2A2;P0DN37</span></span>
<span><span class="co">#&gt; 3     GATD3 A0A0B4J2D5    GATD3;GATD3B A0A0B4J2D5;P0DPI2</span></span>
<span><span class="co">#&gt; 4   PIGBOS1 A0A0B4J2F0         PIGBOS1        A0A0B4J2F0</span></span>
<span><span class="co">#&gt; 5      SIK1 A0A0B4J2F2      SIK1;SIK1B A0A0B4J2F2;P57059</span></span>
<span><span class="co">#&gt; 6   CENPVL1 A0A0U1RR11 CENPVL1;CENPVL2 A0A0U1RR11;P0DPI3</span></span></code></pre></div>
<p><strong>Convert SE</strong>. The expression columns is directly named
by the file names of samples. We recommend to renamed the columns or
renamed the MS files before DIAN-NN search.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Expression col in the DIA-NN report.pg_matrix</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">".raw$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_diann_pg</span><span class="op">)</span>,value <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">ecols</span></span>
<span><span class="co">#&gt; [1] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_A6_1.raw"</span></span>
<span><span class="co">#&gt; [2] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_A6_2.raw"</span></span>
<span><span class="co">#&gt; [3] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_A6_3.raw"</span></span>
<span><span class="co">#&gt; [4] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_B6_1.raw"</span></span>
<span><span class="co">#&gt; [5] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_B6_2.raw"</span></span>
<span><span class="co">#&gt; [6] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.Á_FASP_DIA_B6_3.raw"</span></span>
<span></span>
<span><span class="co">## Convert SE with expdesign table</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se_parse.html">make_se_parse</a></span><span class="op">(</span><span class="va">unique_diann_pg</span>, </span>
<span>                    columns <span class="op">=</span> <span class="va">ecols</span>, </span>
<span>                    mode <span class="op">=</span> <span class="st">"delim"</span>,</span>
<span>                    remove_prefix <span class="op">=</span> <span class="cn">TRUE</span>,   <span class="co">## remove the file prefix </span></span>
<span>                    remove_suffix <span class="op">=</span> <span class="cn">TRUE</span>    <span class="co">## remove the col suffix ".raw"</span></span>
<span>                    <span class="op">)</span></span>
<span></span>
<span><span class="co">## The file prefix and suffix is removed in SE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">#&gt;            label          ID   condition   replicate</span></span>
<span><span class="co">#&gt;      &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; A6_1        A6_1        A6_1          A6           1</span></span>
<span><span class="co">#&gt; A6_2        A6_2        A6_2          A6           2</span></span>
<span><span class="co">#&gt; A6_3        A6_3        A6_3          A6           3</span></span>
<span><span class="co">#&gt; B6_1        B6_1        B6_1          B6           1</span></span>
<span><span class="co">#&gt; B6_2        B6_2        B6_2          B6           2</span></span>
<span><span class="co">#&gt; B6_3        B6_3        B6_3          B6           3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A6_1" "A6_2" "A6_3" "B6_1" "B6_2" "B6_3"</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="long-table-output">Long table output<a class="anchor" aria-label="anchor" href="#long-table-output"></a>
</h4>
<p>Unlike the above cases, the <em>report.tsv</em> from DIA-NN is in
long-format. <code>reshape_long2wide</code> can reshape long tables to
wide-format, before SE construct.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load combined_protein.tsv table</span></span>
<span><span class="va">Diann_repfile</span> <span class="op">&lt;-</span> <span class="st">"./A_spikeIn/Diann_output/report.tsv.gz"</span></span>
<span></span>
<span><span class="va">Diann_rep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">Diann_repfile</span>,sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span> <span class="co">#</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Diann_rep</span><span class="op">)</span> <span class="co"># 390479 rows, each row is a precursor.</span></span>
<span><span class="co">#&gt; [1] 390479     64</span></span>
<span></span>
<span><span class="co">## report.tsv store both Precursor- and PG-level quality</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">Diann_rep</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "File.Name"                 "Run"                      </span></span>
<span><span class="co">#&gt;  [3] "Protein.Group"             "Protein.Ids"              </span></span>
<span><span class="co">#&gt;  [5] "Protein.Names"             "Genes"                    </span></span>
<span><span class="co">#&gt;  [7] "PG.Quantity"               "PG.Normalised"            </span></span>
<span><span class="co">#&gt;  [9] "PG.MaxLFQ"                 "Genes.Quantity"           </span></span>
<span><span class="co">#&gt; [11] "Genes.Normalised"          "Genes.MaxLFQ"             </span></span>
<span><span class="co">#&gt; [13] "Genes.MaxLFQ.Unique"       "Modified.Sequence"        </span></span>
<span><span class="co">#&gt; [15] "Stripped.Sequence"         "Precursor.Id"             </span></span>
<span><span class="co">#&gt; [17] "Precursor.Charge"          "Q.Value"                  </span></span>
<span><span class="co">#&gt; [19] "PEP"                       "Global.Q.Value"           </span></span>
<span><span class="co">#&gt; [21] "Protein.Q.Value"           "PG.Q.Value"               </span></span>
<span><span class="co">#&gt; [23] "Global.PG.Q.Value"         "GG.Q.Value"               </span></span>
<span><span class="co">#&gt; [25] "Translated.Q.Value"        "Proteotypic"              </span></span>
<span><span class="co">#&gt; [27] "Precursor.Quantity"        "Precursor.Normalised"     </span></span>
<span><span class="co">#&gt; [29] "Precursor.Translated"      "Translated.Quality"       </span></span>
<span><span class="co">#&gt; [31] "Ms1.Translated"            "Quantity.Quality"         </span></span>
<span><span class="co">#&gt; [33] "RT"                        "RT.Start"                 </span></span>
<span><span class="co">#&gt; [35] "RT.Stop"                   "iRT"                      </span></span>
<span><span class="co">#&gt; [37] "Predicted.RT"              "Predicted.iRT"            </span></span>
<span><span class="co">#&gt; [39] "First.Protein.Description" "Lib.Q.Value"              </span></span>
<span><span class="co">#&gt; [41] "Lib.PG.Q.Value"            "Ms1.Profile.Corr"         </span></span>
<span><span class="co">#&gt; [43] "Ms1.Area"                  "Evidence"                 </span></span>
<span><span class="co">#&gt; [45] "Spectrum.Similarity"       "Averagine"                </span></span>
<span><span class="co">#&gt; [47] "Mass.Evidence"             "CScore"                   </span></span>
<span><span class="co">#&gt; [49] "Decoy.Evidence"            "Decoy.CScore"             </span></span>
<span><span class="co">#&gt; [51] "Fragment.Quant.Raw"        "Fragment.Quant.Corrected" </span></span>
<span><span class="co">#&gt; [53] "Fragment.Correlations"     "MS2.Scan"                 </span></span>
<span><span class="co">#&gt; [55] "PTM.Informative"           "PTM.Specific"             </span></span>
<span><span class="co">#&gt; [57] "PTM.Localising"            "PTM.Q.Value"              </span></span>
<span><span class="co">#&gt; [59] "PTM.Site.Confidence"       "Lib.PTM.Site.Confidence"  </span></span>
<span><span class="co">#&gt; [61] "IM"                        "iIM"                      </span></span>
<span><span class="co">#&gt; [63] "Predicted.IM"              "Predicted.iIM"</span></span>
<span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Diann_rep</span><span class="op">)</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">T</span>,pageLength <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ae15ba515feda746ad5c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ae15ba515feda746ad5c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_2.raw","F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_3.raw","F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_1.raw","F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_2.raw","F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_3.raw","F:\\DIA_5\\20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_1.raw"],["20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_2","20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_3","20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_1","20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_2","20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_B6_3","20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5&lt;a6&gt;&lt;c1&gt;_FASP_DIA_A6_1"],["P0CG40","P0CG40","P0CG40","P0CG40","P0CG40","P55011"],["P0CG40","P0CG40","P0CG40","P0CG40","P0CG40","P55011"],["SP9_HUMAN","SP9_HUMAN","SP9_HUMAN","SP9_HUMAN","SP9_HUMAN","S12A2_HUMAN"],["SP9","SP9","SP9","SP9","SP9","SLC12A2"],[507866,554254,650874,668742,751487,3288700],[502585,511578,602943,669634,798487,3105380],[502585,511578,602944,669634,798487,2593340],[507866,554254,650874,668742,751487,3288700],[502585,511578,602943,669634,798487,3105380],[502585,511578,602944,669634,798487,2593340],[502585,511578,602944,669634,798487,2593340],["AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAGAGAGAK"],["AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAASAGGKEAASGPNDS","AAAAAAAAAAAAAAAGAGAGAK"],["AAAAAAAAAAAAAAAASAGGKEAASGPNDS4","AAAAAAAAAAAAAAAASAGGKEAASGPNDS4","AAAAAAAAAAAAAAAASAGGKEAASGPNDS4","AAAAAAAAAAAAAAAASAGGKEAASGPNDS4","AAAAAAAAAAAAAAAASAGGKEAASGPNDS4","AAAAAAAAAAAAAAAGAGAGAK3"],[4,4,4,4,4,3],[0.00290477,0.000119093,0.000787349,0.000976288,0.00168301,2.66475e-05],[0.351849,0.00171867,0.0549855,0.0721479,0.165125,0.000441274],[0.0374835,0.0374835,0.0374835,0.0374835,0.0374835,0.000258474],[0.0206249,0.0201483,0.0202929,0.021979,0.0194679,0.000139237],[0.0285883,0.0264217,0.027903,0.0293692,0.0275962,0.000133743],[0.0617671,0.0617671,0.0617671,0.0617671,0.0617671,0.000123716],[0.0290471,0.0270596,0.0277028,0.0294682,0.0280894,0.000135962],[0,0,0,0,0,0],[1,1,1,1,1,1],[507866,554254,650874,668742,751487,1081300],[502585,511578,602943,669634,798487,1056810],[498646,540802,650407,647227,777319,953859],[null,null,null,null,null,null],[0,0,0,0,0,1740560],[0.905064,0.95972,0.897333,0.947106,0.941247,0.995975],[59.1664,58.6064,59.0602,58.7336,58.6171,65.0505],[58.93,58.3697,58.8242,58.4971,58.3795,64.8126],[59.4035,58.8432,59.2967,58.9702,58.8545,65.2888],[34.2568,34.2568,34.2568,34.2568,34.2568,41.8375],[58.9905,58.664,59.0449,58.6997,58.6116,65.3254],[34.4241,34.1716,34.2902,34.2369,34.2617,41.4578],["Transcription factor Sp9","Transcription factor Sp9","Transcription factor Sp9","Transcription factor Sp9","Transcription factor Sp9","Solute carrier family 12 member 2"],[0.00538048,0.00538048,0.00538048,0.00538048,0.00538048,3.75601e-05],[0.0325691,0.0325691,0.0325691,0.0325691,0.0325691,0.000129383],[0,0,0,0,0,0.979495],[0,0,0,0,0,1973110],[1.50343,3.44797,1.85121,2.4184,2.51614,6.75941],[1.03191e-14,0.520573,0.0430605,0.373779,0.467442,0.91093],[0.0378933,0.0378934,0.0378934,0.0378934,0.0378934,1],[0,0,0,0,0,0],[0.859137,0.999184,0.974438,0.966816,0.928522,0.999795],[2.87035,3.1243,3.67111,3.70096,3.60648,1.11636],[0.866007,0.88809,0.801329,0.966797,0.84182,0.0441394],["71166.7;0;18413.9;12018.8;299097;137603;118273;104208;27405.4;0;0;0;","117899;5209.53;4747.19;62924.8;268649;167706;78620.5;121462;53270.3;0;0;0;","160732;0;20341;12698.9;272323;217819;119979;214770;44603.8;0;0;0;","174129;0;0;28980.5;293972;200642;127144;212351;26928.9;0;0;0;","195358;0;16393.4;59014.9;346000;210128;109588;209135;59271.2;0;0;0;","827234;367664;366917;363881;350505;312323;324379;326154;256538;236549;166755;168704;"],["71166.7;0;18413.9;12018.8;299097;137603;118273;104208;27405.4;0;0;0;","117899;5209.53;4747.19;62924.8;268649;167706;78620.5;121462;53270.3;0;0;0;","160732;0;20341;12698.9;272323;217819;119979;214770;44603.8;0;0;0;","174129;0;0;28980.5;293972;200642;127144;212351;26928.9;0;0;0;","195358;0;16393.4;59014.9;346000;210128;109588;209135;59271.2;0;0;0;","827234;367664;366917;363881;350505;312323;324379;326154;256538;236549;166755;168704;"],["0.596265;0;0.481072;0.469287;0.992117;0.87555;0.95262;0.879947;0.690505;0;0;0;","0.946358;0.538548;0.538548;0.991095;0.985745;0.927424;0.760696;0.981553;0.895773;0;0;0;","0.903412;0;0.341617;0.428567;0.815648;0.994973;0.783636;0.970016;0.564806;0;0;0;","0.995275;0;0;0.54288;0.890857;0.987716;0.539425;0.96081;0.6632;0;0;0;","0.994446;0;0.217862;0.762787;0.888472;0.978688;0.645339;0.969931;0.607064;0;0;0;","0.97976;0.997339;0.995379;0.997228;0.995297;0.993732;0.997043;0.992573;0.99627;0.996218;0.993245;0.983539;"],[40964,40586,40922,40670,40586,45075],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0.99462,0.945014,0.927852,0,0.999559],[0,0,0,0,0,0.999547],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>File.Name<\/th>\n      <th>Run<\/th>\n      <th>Protein.Group<\/th>\n      <th>Protein.Ids<\/th>\n      <th>Protein.Names<\/th>\n      <th>Genes<\/th>\n      <th>PG.Quantity<\/th>\n      <th>PG.Normalised<\/th>\n      <th>PG.MaxLFQ<\/th>\n      <th>Genes.Quantity<\/th>\n      <th>Genes.Normalised<\/th>\n      <th>Genes.MaxLFQ<\/th>\n      <th>Genes.MaxLFQ.Unique<\/th>\n      <th>Modified.Sequence<\/th>\n      <th>Stripped.Sequence<\/th>\n      <th>Precursor.Id<\/th>\n      <th>Precursor.Charge<\/th>\n      <th>Q.Value<\/th>\n      <th>PEP<\/th>\n      <th>Global.Q.Value<\/th>\n      <th>Protein.Q.Value<\/th>\n      <th>PG.Q.Value<\/th>\n      <th>Global.PG.Q.Value<\/th>\n      <th>GG.Q.Value<\/th>\n      <th>Translated.Q.Value<\/th>\n      <th>Proteotypic<\/th>\n      <th>Precursor.Quantity<\/th>\n      <th>Precursor.Normalised<\/th>\n      <th>Precursor.Translated<\/th>\n      <th>Translated.Quality<\/th>\n      <th>Ms1.Translated<\/th>\n      <th>Quantity.Quality<\/th>\n      <th>RT<\/th>\n      <th>RT.Start<\/th>\n      <th>RT.Stop<\/th>\n      <th>iRT<\/th>\n      <th>Predicted.RT<\/th>\n      <th>Predicted.iRT<\/th>\n      <th>First.Protein.Description<\/th>\n      <th>Lib.Q.Value<\/th>\n      <th>Lib.PG.Q.Value<\/th>\n      <th>Ms1.Profile.Corr<\/th>\n      <th>Ms1.Area<\/th>\n      <th>Evidence<\/th>\n      <th>Spectrum.Similarity<\/th>\n      <th>Averagine<\/th>\n      <th>Mass.Evidence<\/th>\n      <th>CScore<\/th>\n      <th>Decoy.Evidence<\/th>\n      <th>Decoy.CScore<\/th>\n      <th>Fragment.Quant.Raw<\/th>\n      <th>Fragment.Quant.Corrected<\/th>\n      <th>Fragment.Correlations<\/th>\n      <th>MS2.Scan<\/th>\n      <th>PTM.Informative<\/th>\n      <th>PTM.Specific<\/th>\n      <th>PTM.Localising<\/th>\n      <th>PTM.Q.Value<\/th>\n      <th>PTM.Site.Confidence<\/th>\n      <th>Lib.PTM.Site.Confidence<\/th>\n      <th>IM<\/th>\n      <th>iIM<\/th>\n      <th>Predicted.IM<\/th>\n      <th>Predicted.iIM<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":6,"columnDefs":[{"className":"dt-right","targets":[7,8,9,10,11,12,13,17,18,19,20,21,22,23,24,25,26,27,28,29,31,32,33,34,35,36,37,38,40,41,42,43,44,45,46,47,48,49,50,54,55,56,57,58,59,60,61,62,63,64]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[6,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><p><strong>Reshape</strong> long-table.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Diann_rep_wided</span> <span class="op">=</span> <span class="fu"><a href="../reference/reshape_long2wide.html">reshape_long2wide</a></span><span class="op">(</span><span class="va">Diann_rep</span>,</span>
<span>                                    sample_col <span class="op">=</span> <span class="st">"File.Name"</span>,      <span class="co"># the column labeling sample names</span></span>
<span>                                    feature_col <span class="op">=</span> <span class="st">"Protein.Group"</span>, <span class="co"># PG IDs</span></span>
<span>                                    expression_col <span class="op">=</span> <span class="st">"PG.MaxLFQ"</span>,  <span class="co"># PG quantity. Normalized one is also ok。</span></span>
<span>                                    shrink_ident_cols  <span class="op">=</span> <span class="st">"Genes"</span>,  <span class="co"># Gene names</span></span>
<span>                                    extend_ident_cols <span class="op">=</span>  <span class="st">"Protein.Q.Value"</span>,  <span class="co"># optional, some identification info.</span></span>
<span>                                    remove_sample_prefix <span class="op">=</span> <span class="cn">FALSE</span>,  <span class="co"># remove prefix in sample_col</span></span>
<span>                                    remove_sample_suffix <span class="op">=</span> <span class="cn">FALSE</span>   <span class="co"># remove suffix in sample_col</span></span>
<span>                                    <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Diann_rep_wided</span><span class="op">)</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">T</span>,pageLength <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-7a49959142412b299a78" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7a49959142412b299a78">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["A0A023PYJ4","A0A023PZB8","A0A087X1C5;P10635","A0A0A6YYC5","A0A0B4J2A2;P0DN37","A0A0B4J2D5;P0DPI2"],[256417,210148,null,204537,7640030,628954],[212780,177478,null,186357,7062470,603493],[292507,198770,143270,189840,7274770,614230],[298445,152864,88324.8,257882,7412950,581705],[259358,191724,128566,283105,7364290,543701],[94499.5,140902,59624.8,157295,6093270,403160],["YMR153C-A","YER079C-A","CYP2D6;CYP2D7","TRAV14DV4","PPIAL4C;PPIAL4G","GATD3;GATD3B"],[0.0204673,0.00811567,null,0.0168784,1,1],[0.0200768,0.00801865,null,0.0174136,1,1],[0.020219,0.00822814,1,0.016881,1,1],[0.021979,0.00762281,1,0.0179512,1,1],[0.0194679,0.0124591,1,0.0172214,1,1],[0.0195975,0.00824047,1,0.01911,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Protein.Group<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_2.raw<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_3.raw<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_1.raw<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_2.raw<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_3.raw<\/th>\n      <th>F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_1.raw<\/th>\n      <th>Genes<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_2.raw<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_3.raw<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_1.raw<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_2.raw<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_3.raw<\/th>\n      <th>Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_1.raw<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":6,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,9,10,11,12,13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[6,10,25,50,100]}},"evals":[],"jsHooks":[]}</script><p><strong>Unique identifier</strong>. The <em>names</em> and
<em>Protein.Group</em> are in the columns “Genes” and
“Protein.Group”.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate unique names and ids</span></span>
<span><span class="va">unique_diann_pg2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">Diann_rep_wided</span>, </span>
<span>                                name <span class="op">=</span> <span class="st">"Genes"</span>,  <span class="co">#gene </span></span>
<span>                                ids <span class="op">=</span> <span class="st">"Protein.Group"</span>   <span class="co">#protein</span></span>
<span>                                <span class="op">)</span></span>
<span></span>
<span><span class="co">## new columns name and id, which is necessary to make_se</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">unique_diann_pg2</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"ID"</span>, <span class="st">"Genes"</span>, <span class="st">"Protein.Group"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;        name         ID           Genes     Protein.Group</span></span>
<span><span class="co">#&gt; 1 YMR153C-A A0A023PYJ4       YMR153C-A        A0A023PYJ4</span></span>
<span><span class="co">#&gt; 2 YER079C-A A0A023PZB8       YER079C-A        A0A023PZB8</span></span>
<span><span class="co">#&gt; 3    CYP2D6 A0A087X1C5   CYP2D6;CYP2D7 A0A087X1C5;P10635</span></span>
<span><span class="co">#&gt; 4 TRAV14DV4 A0A0A6YYC5       TRAV14DV4        A0A0A6YYC5</span></span>
<span><span class="co">#&gt; 5   PPIAL4C A0A0B4J2A2 PPIAL4C;PPIAL4G A0A0B4J2A2;P0DN37</span></span>
<span><span class="co">#&gt; 6     GATD3 A0A0B4J2D5    GATD3;GATD3B A0A0B4J2D5;P0DPI2</span></span></code></pre></div>
<p><strong>Convert SE</strong>. The expression columns is directly named
by the file names of samples.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Expression col in the DIA-NN report.pg_matrix</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">".raw$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_diann_pg2</span><span class="op">)</span>,value <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">ecols</span> <span class="co"># contains Protein.Q.Value columns</span></span>
<span><span class="co">#&gt;  [1] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_2.raw"                </span></span>
<span><span class="co">#&gt;  [2] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_3.raw"                </span></span>
<span><span class="co">#&gt;  [3] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_1.raw"                </span></span>
<span><span class="co">#&gt;  [4] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_2.raw"                </span></span>
<span><span class="co">#&gt;  [5] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_3.raw"                </span></span>
<span><span class="co">#&gt;  [6] "F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_1.raw"                </span></span>
<span><span class="co">#&gt;  [7] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_2.raw"</span></span>
<span><span class="co">#&gt;  [8] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_3.raw"</span></span>
<span><span class="co">#&gt;  [9] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_1.raw"</span></span>
<span><span class="co">#&gt; [10] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_2.raw"</span></span>
<span><span class="co">#&gt; [11] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_B6_3.raw"</span></span>
<span><span class="co">#&gt; [12] "Protein.Q.Value.F..DIA_5.20211207_LM_SA_ZXF_LY_FZH_Hela_AH109_DH5.._FASP_DIA_A6_1.raw"</span></span>
<span></span>
<span><span class="va">ecols</span> <span class="op">=</span> <span class="va">ecols</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## Convert SE with expdesign table</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se_parse.html">make_se_parse</a></span><span class="op">(</span><span class="va">unique_diann_pg2</span>, </span>
<span>                    columns <span class="op">=</span> <span class="va">ecols</span>, </span>
<span>                    mode <span class="op">=</span> <span class="st">"delim"</span>,</span>
<span>                    remove_prefix <span class="op">=</span> <span class="cn">TRUE</span>,   <span class="co">## remove the file prefix </span></span>
<span>                    remove_suffix <span class="op">=</span> <span class="cn">TRUE</span>    <span class="co">## remove the col suffix ".raw"</span></span>
<span>                    <span class="op">)</span></span>
<span></span>
<span><span class="co">## The file prefix and suffix is removed in SE</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">#&gt;            label          ID   condition   replicate</span></span>
<span><span class="co">#&gt;      &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; A6_2        A6_2        A6_2          A6           2</span></span>
<span><span class="co">#&gt; A6_3        A6_3        A6_3          A6           3</span></span>
<span><span class="co">#&gt; B6_1        B6_1        B6_1          B6           1</span></span>
<span><span class="co">#&gt; B6_2        B6_2        B6_2          B6           2</span></span>
<span><span class="co">#&gt; B6_3        B6_3        B6_3          B6           3</span></span>
<span><span class="co">#&gt; A6_1        A6_1        A6_1          A6           1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A6_2" "A6_3" "B6_1" "B6_2" "B6_3" "A6_1"</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19043)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                               </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] DEP2_0.2.05.23              R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [3] limma_3.54.1                MSnbase_2.24.2             </span></span>
<span><span class="co">#&gt;  [5] ProtGenerics_1.30.0         mzR_2.32.0                 </span></span>
<span><span class="co">#&gt;  [7] Rcpp_1.0.10                 MsCoreUtils_1.10.0         </span></span>
<span><span class="co">#&gt;  [9] SummarizedExperiment_1.28.0 Biobase_2.58.0             </span></span>
<span><span class="co">#&gt; [11] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        </span></span>
<span><span class="co">#&gt; [13] IRanges_2.32.0              S4Vectors_0.36.1           </span></span>
<span><span class="co">#&gt; [15] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      </span></span>
<span><span class="co">#&gt; [17] matrixStats_0.63.0          BiocStyle_2.26.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] circlize_0.4.15             systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt;   [3] plyr_1.8.8                  igraph_1.4.0               </span></span>
<span><span class="co">#&gt;   [5] lazyeval_0.2.2              splines_4.2.1              </span></span>
<span><span class="co">#&gt;   [7] crosstalk_1.2.0             BiocParallel_1.32.5        </span></span>
<span><span class="co">#&gt;   [9] TCseq_1.22.6                ggplot2_3.4.1              </span></span>
<span><span class="co">#&gt;  [11] digest_0.6.31               foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.4             fansi_1.0.4                </span></span>
<span><span class="co">#&gt;  [15] magrittr_2.0.3              memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [17] cluster_2.1.4               doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [19] ComplexHeatmap_2.14.0       Biostrings_2.66.0          </span></span>
<span><span class="co">#&gt;  [21] annotate_1.76.0             askpass_1.1                </span></span>
<span><span class="co">#&gt;  [23] pkgdown_2.0.7               colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [25] blob_1.2.3                  textshaping_0.3.6          </span></span>
<span><span class="co">#&gt;  [27] xfun_0.39                   dplyr_1.1.0                </span></span>
<span><span class="co">#&gt;  [29] crayon_1.5.2                RCurl_1.98-1.10            </span></span>
<span><span class="co">#&gt;  [31] jsonlite_1.8.4              missForest_1.5             </span></span>
<span><span class="co">#&gt;  [33] impute_1.72.3               survival_3.5-3             </span></span>
<span><span class="co">#&gt;  [35] iterators_1.0.14            glue_1.6.2                 </span></span>
<span><span class="co">#&gt;  [37] gtable_0.3.1                zlibbioc_1.44.0            </span></span>
<span><span class="co">#&gt;  [39] XVector_0.38.0              GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;  [41] DelayedArray_0.23.2         shape_1.4.6                </span></span>
<span><span class="co">#&gt;  [43] scales_1.2.1                vsn_3.66.0                 </span></span>
<span><span class="co">#&gt;  [45] rngtools_1.5.2              DBI_1.1.3                  </span></span>
<span><span class="co">#&gt;  [47] edgeR_3.40.2                xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [49] clue_0.3-64                 reticulate_1.28            </span></span>
<span><span class="co">#&gt;  [51] bit_4.0.5                   proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [53] preprocessCore_1.60.2       DT_0.27                    </span></span>
<span><span class="co">#&gt;  [55] umap_0.2.10.0               glmnet_4.1-6               </span></span>
<span><span class="co">#&gt;  [57] htmlwidgets_1.6.1           httr_1.4.4                 </span></span>
<span><span class="co">#&gt;  [59] RColorBrewer_1.1-3          ellipsis_0.3.2             </span></span>
<span><span class="co">#&gt;  [61] pkgconfig_2.0.3             XML_3.99-0.13              </span></span>
<span><span class="co">#&gt;  [63] sass_0.4.5                  locfit_1.5-9.7             </span></span>
<span><span class="co">#&gt;  [65] utf8_1.2.3                  tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;  [67] rlang_1.0.6                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [69] AnnotationDbi_1.60.0        munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [71] tools_4.2.1                 cachem_1.0.6               </span></span>
<span><span class="co">#&gt;  [73] cli_3.6.0                   generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [75] RSQLite_2.3.0               QFeatures_1.8.0            </span></span>
<span><span class="co">#&gt;  [77] fdrtool_1.2.17              evaluate_0.20              </span></span>
<span><span class="co">#&gt;  [79] stringr_1.5.0               fastmap_1.1.0              </span></span>
<span><span class="co">#&gt;  [81] mzID_1.36.0                 yaml_2.3.7                 </span></span>
<span><span class="co">#&gt;  [83] ragg_1.2.5                  knitr_1.42                 </span></span>
<span><span class="co">#&gt;  [85] bit64_4.0.5                 fs_1.6.1                   </span></span>
<span><span class="co">#&gt;  [87] purrr_1.0.1                 randomForest_4.7-1.1       </span></span>
<span><span class="co">#&gt;  [89] KEGGREST_1.38.0             AnnotationFilter_1.22.0    </span></span>
<span><span class="co">#&gt;  [91] ncdf4_1.21                  doRNG_1.8.6                </span></span>
<span><span class="co">#&gt;  [93] compiler_4.2.1              rstudioapi_0.14            </span></span>
<span><span class="co">#&gt;  [95] png_0.1-8                   e1071_1.7-13               </span></span>
<span><span class="co">#&gt;  [97] affyio_1.68.0               tibble_3.1.8               </span></span>
<span><span class="co">#&gt;  [99] geneplotter_1.76.0          bslib_0.4.2                </span></span>
<span><span class="co">#&gt; [101] stringi_1.7.12              RSpectra_0.16-1            </span></span>
<span><span class="co">#&gt; [103] desc_1.4.2                  lattice_0.20-45            </span></span>
<span><span class="co">#&gt; [105] Matrix_1.5-3                vctrs_0.5.2                </span></span>
<span><span class="co">#&gt; [107] pillar_1.8.1                lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt; [109] BiocManager_1.30.19         jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [111] MALDIquant_1.22             GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [113] data.table_1.14.8           bitops_1.0-7               </span></span>
<span><span class="co">#&gt; [115] pcaMethods_1.90.0           affy_1.76.0                </span></span>
<span><span class="co">#&gt; [117] bookdown_0.34               codetools_0.2-19           </span></span>
<span><span class="co">#&gt; [119] MASS_7.3-58.2               assertthat_0.2.1           </span></span>
<span><span class="co">#&gt; [121] openssl_2.0.5               DESeq2_1.38.3              </span></span>
<span><span class="co">#&gt; [123] rprojroot_2.0.3             rjson_0.2.21               </span></span>
<span><span class="co">#&gt; [125] withr_2.5.0                 GenomicAlignments_1.34.0   </span></span>
<span><span class="co">#&gt; [127] Rsamtools_2.14.0            GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">#&gt; [129] parallel_4.2.1              MultiAssayExperiment_1.24.0</span></span>
<span><span class="co">#&gt; [131] grid_4.2.1                  tidyr_1.3.0                </span></span>
<span><span class="co">#&gt; [133] class_7.3-21                rmarkdown_2.20             </span></span>
<span><span class="co">#&gt; [135] Rtsne_0.16                  itertools_0.1-3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Zhenhuan Feng, Xiaofei Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
