<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="DEP2">
<title>Quick start • DEP2</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Quick start">
<meta property="og:description" content="DEP2">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DEP2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.06.12</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/app_demo.html">DEP2 Application Guide</a>
    <a class="dropdown-item" href="../articles/Data_import.html">Data import</a>
    <a class="dropdown-item" href="../articles/DEP2_analysis2.html">DEP2 analysis workflow</a>
    <a class="dropdown-item" href="../articles/Quick_start.html">Quick start</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Quick start</h1>
                        <h4 data-toc-skip class="author">Zhenhuan
Feng</h4>
                        
      
      
      <div class="d-none name"><code>Quick_start.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vigenette gave an outline of DEP2 proteomics analysis
      pipeline.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>DEP2 provides a differentially expressed/enriched analysis toolkit
for mass spectrometry based quantitative proteomics data via limma <span class="citation">(Ritchie et al. 2015)</span>, upgraded from a previous
package DEP <span class="citation">(Zhang et al. 2018)</span>. This
package provides an integrated workflow for proteomics analysis,
including data processing, missing value imputation, hypothesis test,
visualization and downstream functional exploration.</p>
<p>Now, DEP2 provide three types of differential proteomics
analysis:</p>
<ol style="list-style-type: decimal">
<li><p>The pipeline for proteins-level expression/enrichment analysis,
started from protein-level quantity.</p></li>
<li><p>The pipeline for proteins-level expression/enrichment analysis,
started from peptide-level quantity.</p></li>
<li><p>The pipeline for post-translation modification (PTM) specified
proteomics, performed upon modified peptides quantity.</p></li>
</ol>
<p>Besides, DEP2 also packaged a RNA-seq data analysis pipeline from
<em><a href="https://bioconductor.org/packages/3.16/DESeq2" class="external-link">DESeq2</a></em>.</p>
</div>
<div class="section level2">
<h2 id="load-required-packages">Load required packages<a class="anchor" aria-label="anchor" href="#load-required-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEP2</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>We demonstrate DEP2 pipelines using the data from a multiple-omics
study of silicosis mouse model(Wang et al. 2022), which are built-in
data in the package. The data in table format is also available from the
repository (github.com/mildpiggy/OmicsExample).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The proteomics data (proteingroup.txt output from MaxQuant).</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_pg</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## The proteomics data in peptide-level (peptides.txt output from MaxQuant).</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_peptide</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## The the experiment design table annotating samples.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_ExpDesign</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The Phosphoproteomics data (Phospho(STY)Sites.txt output from MaxQuant).</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_phos</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">## The RNA-seq counts data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_counts</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-start">Get start<a class="anchor" aria-label="anchor" href="#get-start"></a>
</h2>
<p>The four pipeline can perform step by step as follow</p>
<div class="section level3">
<h3 id="proteingroup-workflow">Proteingroup workflow<a class="anchor" aria-label="anchor" href="#proteingroup-workflow"></a>
</h3>
<p>The conventional workflow began from protein-level data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 1. Construct SummarizedExperiement</span></span>
<span><span class="va">unique_pg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">Silicosis_pg</span>, names <span class="op">=</span> <span class="st">"Gene.names"</span>, ids <span class="op">=</span> <span class="st">"Protein.IDs"</span><span class="op">)</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"LFQ.intensity."</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_pg</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se.html">make_se</a></span><span class="op">(</span><span class="va">unique_pg</span>,                       </span>
<span>              columns <span class="op">=</span> <span class="va">ecols</span>, </span>
<span>              expdesign <span class="op">=</span> <span class="va">Silicosis_ExpDesign</span></span>
<span>             <span class="op">)</span></span>
<span></span>
<span><span class="co">## 2. Filter</span></span>
<span><span class="va">filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_se.html">filter_se</a></span><span class="op">(</span><span class="va">se</span>,</span>
<span>                  thr <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  filter_formula <span class="op">=</span> <span class="op">~</span> <span class="va">Reverse</span> <span class="op">!=</span> <span class="st">'+'</span> <span class="op">&amp;</span> <span class="va">Potential.contaminant</span> <span class="op">!=</span><span class="st">"+"</span></span>
<span>                  <span class="op">)</span></span>
<span></span>
<span><span class="co">## 3. Normalize</span></span>
<span><span class="va">norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_vsn.html">normalize_vsn</a></span><span class="op">(</span><span class="va">filt</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 4. Impute</span></span>
<span><span class="va">filt</span> <span class="op">&lt;-</span> <span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/impute.html">impute</a></span><span class="op">(</span><span class="va">norm</span>,fun <span class="op">=</span> <span class="st">"MinDet"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 5. Differential test</span></span>
<span><span class="va">diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/test_diff.html">test_diff</a></span><span class="op">(</span><span class="va">filt</span>,type <span class="op">=</span> <span class="st">"control"</span>, control <span class="op">=</span> <span class="st">"PBS"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEP2_add_rejections.html">add_rejections</a></span><span class="op">(</span><span class="va">diff</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, lfc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ptm-peptides-workflow">PTM-peptides workflow<a class="anchor" aria-label="anchor" href="#ptm-peptides-workflow"></a>
</h3>
<p>The workflow for modified peptides data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 1. Construct SummarizedExperiement</span></span>
<span><span class="va">unique_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_unique_ptm.html">make_unique_ptm</a></span><span class="op">(</span><span class="va">Silicosis_phos</span>, </span>
<span>                              gene_name <span class="op">=</span> <span class="st">"Gene.names"</span>, protein_ID <span class="op">=</span> <span class="st">"Protein"</span>,</span>
<span>                              aa <span class="op">=</span> <span class="st">"Amino.acid"</span>, pos <span class="op">=</span> <span class="st">"Position"</span><span class="op">)</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">unique_ptm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">se_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_se.html">make_se</a></span><span class="op">(</span><span class="va">unique_ptm</span>,                       </span>
<span>                  columns <span class="op">=</span> <span class="va">ecols</span>, </span>
<span>                  expdesign <span class="op">=</span> <span class="va">Silicosis_ExpDesign</span></span>
<span>                  <span class="op">)</span></span>
<span></span>
<span><span class="co">## 2. Filter</span></span>
<span><span class="va">filt_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_se.html">filter_se</a></span><span class="op">(</span><span class="va">se_ptm</span>,</span>
<span>                     thr <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                     filter_formula <span class="op">=</span> <span class="op">~</span> <span class="va">Reverse</span><span class="op">!=</span><span class="st">"+"</span> <span class="op">&amp;</span> </span>
<span>                       <span class="va">Potential.contaminant</span><span class="op">!=</span><span class="st">"+"</span> <span class="op">&amp;</span> </span>
<span>                       <span class="va">Localization.prob</span> <span class="op">&gt;</span> <span class="fl">0.7</span></span>
<span>                     <span class="op">)</span></span>
<span></span>
<span><span class="co">## 3. Normalize</span></span>
<span><span class="va">norm_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_vsn.html">normalize_vsn</a></span><span class="op">(</span><span class="va">filt_ptm</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 4. Impute</span></span>
<span><span class="va">filt_ptm</span> <span class="op">&lt;-</span> <span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/impute.html">impute</a></span><span class="op">(</span><span class="va">norm_ptm</span>, fun <span class="op">=</span> <span class="st">"QRILC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 5. Differential test</span></span>
<span><span class="va">diff_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/test_diff.html">test_diff</a></span><span class="op">(</span><span class="va">filt_ptm</span>,type <span class="op">=</span> <span class="st">"control"</span>, control <span class="op">=</span> <span class="st">"PBS"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dep_ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEP2_add_rejections.html">add_rejections</a></span><span class="op">(</span><span class="va">diff_ptm</span>, alpha <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="peptides-aggregation-workflow">Peptides aggregation workflow<a class="anchor" aria-label="anchor" href="#peptides-aggregation-workflow"></a>
</h3>
<p>The peptide to protein aggregation workflow.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 1. Construct SummarizedExperiement</span></span>
<span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Intensity."</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">Silicosis_peptide</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">pe</span> <span class="op">=</span> <span class="fu"><a href="../reference/make_pe_parse.html">make_pe_parse</a></span><span class="op">(</span><span class="va">Silicosis_peptide</span>, </span>
<span>                    columns <span class="op">=</span> <span class="va">ecols</span>,   <span class="co"># columns is the 'Intensity' colunmns</span></span>
<span>                    mode <span class="op">=</span> <span class="st">"delim"</span>, sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>                    remove_prefix <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                   <span class="op">)</span></span>
<span></span>
<span><span class="co">## 2. Filter</span></span>
<span><span class="va">pe</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_pe.html">filter_pe</a></span><span class="op">(</span><span class="va">pe</span>, </span>
<span>               thr <span class="op">=</span> <span class="fl">1</span>,</span>
<span>               fraction <span class="op">=</span> <span class="fl">0.3</span>, </span>
<span>               filter_formula <span class="op">=</span> <span class="op">~</span> <span class="va">Reverse</span> <span class="op">!=</span> <span class="st">'+'</span> <span class="op">&amp;</span> <span class="va">Potential.contaminant</span> <span class="op">!=</span><span class="st">"+"</span> </span>
<span>               <span class="op">)</span></span>
<span></span>
<span><span class="co">## 3. Impute</span></span>
<span><span class="va">pe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute_pe.html">impute_pe</a></span><span class="op">(</span><span class="va">pe</span>, fun <span class="op">=</span> <span class="st">"QRILC"</span>, name <span class="op">=</span> <span class="st">"peptideImp"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 4. Normalize</span></span>
<span><span class="va">pe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_pe.html">normalize_pe</a></span><span class="op">(</span><span class="va">pe</span>,method <span class="op">=</span> <span class="st">"quantiles.robust"</span>, i <span class="op">=</span> <span class="st">"peptideImp"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 5. Aggregate peptides quantity to proteins'</span></span>
<span><span class="va">begin_time</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/aggregate_pe.html">aggregate_pe</a></span><span class="op">(</span><span class="va">pe</span>, fcol <span class="op">=</span> <span class="st">"Proteins"</span>, reserve <span class="op">=</span> <span class="st">"Gene.names"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="va">begin_time</span><span class="op">)</span> <span class="co">## Required few minutes</span></span>
<span></span>
<span><span class="co">## 6. Transform a SummarizedExperiement of protein quantities.</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pe2se.html">pe2se</a></span><span class="op">(</span><span class="va">pe</span>, names <span class="op">=</span> <span class="st">"Gene.names"</span>, ids <span class="op">=</span> <span class="st">"smallestProteingroups"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## 7. Differential test</span></span>
<span><span class="va">diff_pep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/test_diff.html">test_diff</a></span><span class="op">(</span><span class="va">se</span>,type <span class="op">=</span> <span class="st">"control"</span>, control <span class="op">=</span> <span class="st">"PBS"</span><span class="op">)</span></span>
<span><span class="va">dep_pep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEP2_add_rejections.html">add_rejections</a></span><span class="op">(</span><span class="va">diff_pep</span>, alpha <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get_signicant.html">get_signicant</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="rna-seq-workflow">RNA-seq workflow<a class="anchor" aria-label="anchor" href="#rna-seq-workflow"></a>
</h3>
<p>DEP2 packages a simple pipeline from <code>DESeq2</code> for the
RNA-seq data (gene counts).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 0. Check the depend packages</span></span>
<span><span class="fu"><a href="../reference/check_RNAseq_depends.html">check_RNAseq_depends</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span></span>
<span><span class="co">## 1. Construct DESeqDataSet </span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_dds_parse.html">make_dds_parse</a></span><span class="op">(</span><span class="va">Silicosis_counts</span>,mode <span class="op">=</span> <span class="st">"delim"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 2. Filter</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_se.html">filter_se</a></span><span class="op">(</span><span class="va">dds</span>, fraction <span class="op">=</span> <span class="fl">0.3</span>, thr <span class="op">=</span> <span class="fl">1</span>, rowsum_threshold <span class="op">=</span> <span class="fl">35</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## 3. Transform ID (optional)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ID_transform.html">ID_transform</a></span><span class="op">(</span><span class="va">dds</span>, fromtype <span class="op">=</span> <span class="st">"ENSEMBL"</span>, species <span class="op">=</span> <span class="st">"Mouse"</span><span class="op">)</span></span>
<span><span class="co">#&gt; org.Mm.eg.db</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Gnai3" "Cdc45" "Apoh"  "Narf"  "Cav2"  "Klf6"</span></span>
<span></span>
<span><span class="co">## 3. Differential test</span></span>
<span><span class="va">diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/test_diff_deg.html">test_diff_deg</a></span><span class="op">(</span><span class="va">dds</span>, type <span class="op">=</span> <span class="st">"control"</span>, control <span class="op">=</span> <span class="st">"PBS"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Performing DESeq analysis... </span></span>
<span><span class="co">#&gt; Formatting analysis result... </span></span>
<span><span class="co">#&gt; Calulate a rlog &amp; log2 transform assay, stored in rlg or ntd slot...</span></span>
<span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEP2_add_rejections.html">add_rejections</a></span><span class="op">(</span><span class="va">diff</span>, alpha <span class="op">=</span> <span class="fl">0.01</span>, lfc <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="result-visualization-and-table-export">Result visualization and table export<a class="anchor" aria-label="anchor" href="#result-visualization-and-table-export"></a>
</h3>
<p>DEP2 provides functions to visualize differential test results.</p>
<p><code>plot_heatmap</code> can draw a <em>ComplexHeatmap</em> upon SE
or DEGdata object out from <code>add_rejections</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">dep</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "SummarizedExperiment"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "SummarizedExperiment"</span></span>
<span></span>
<span><span class="co">## Sort the conditions</span></span>
<span><span class="va">dep</span> <span class="op">=</span> <span class="fu"><a href="../reference/Order_cols.html">Order_cols</a></span><span class="op">(</span><span class="va">dep</span>, order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PBS"</span>,<span class="st">"W2"</span>,<span class="st">"W4"</span>,<span class="st">"W6"</span>,<span class="st">"W9"</span>,<span class="st">"W10"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Heatmap on centered values</span></span>
<span><span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                   cluster_column <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                   kmeans <span class="op">=</span> <span class="cn">T</span>,        <span class="co"># kmeans cluster rows</span></span>
<span>                   col_limit <span class="op">=</span> <span class="fl">4</span>      <span class="co"># limit of the color scale</span></span>
<span>                   <span class="op">)</span>     </span></code></pre></div>
<p><img src="Quick_start_files/figure-html/heatmap1-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Heatmap on contrast log2 foldchange</span></span>
<span><span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                   type <span class="op">=</span> <span class="st">"contrast"</span>,</span>
<span>                   cluster_column <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                   kmeans <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                   col_limit <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/heatmap1-2.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Manual contrast</span></span>
<span><span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                   manual_contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"W6_vs_PBS"</span>,<span class="st">"W4_vs_PBS"</span><span class="op">)</span>,</span>
<span>                   cluster_column <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                   kmeans <span class="op">=</span> <span class="cn">T</span>,        </span>
<span>                   col_limit <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/heatmap1-3.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Change color panel </span></span>
<span><span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                   type <span class="op">=</span> <span class="st">"contrast"</span>,</span>
<span>                   color <span class="op">=</span> <span class="st">"PiYG"</span>,</span>
<span>                   cluster_column <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                   kmeans <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                   col_limit <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/heatmap1-4.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co">## Parameters in ComplexHeatmap::Heatmap is available</span></span>
<span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu">DEP2</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                         cluster_column <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                         kmeans <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                         cluster_rows <span class="op">=</span> <span class="cn">F</span></span>
<span>                         <span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/heatmap1-5.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Heatmap can be saved as follow</span></span>
<span><span class="co"># pdf("ht.pdf")</span></span>
<span><span class="co"># ht</span></span>
<span><span class="co"># dev.off()</span></span></code></pre></div>
<p>Volcano plot is another common figure in omics study, which can be
drawn by <code>plot_volcano</code> as follow.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">dep</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/volcano2-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">deg</span>, </span>
<span>             contrast <span class="op">=</span> <span class="st">"W6_vs_PBS"</span>,  <span class="co"># contrast</span></span>
<span>             label_number <span class="op">=</span> <span class="fl">20</span></span>
<span>             <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 40 rows containing missing values (`geom_point()`).</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/volcano2-2.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">volc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">deg</span>, contrast <span class="op">=</span> <span class="st">"W4_vs_PBS"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Warning: Removed 40 rows containing missing values (`geom_point()`).</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/volcano2-3.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># ggplot2::ggsave(filename = "volcano.png") #export to file</span></span></code></pre></div>
<p>It is easy to export table from S4 object that inherits SE class.
Functions like <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment::colData</a></code> and
<code>rowData</code> can easily extract message in object. DEP2 contains
<code>get_significant</code> and <code>get_df_wide</code> to generate
significant result table or full dataset from analysis result.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## significant table</span></span>
<span><span class="va">sig_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_signicant.html">get_signicant</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                        return_type <span class="op">=</span> <span class="st">"table"</span> </span>
<span>                        <span class="op">)</span></span>
<span><span class="va">sig_tb_W6_vs_PBS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_signicant.html">get_signicant</a></span><span class="op">(</span><span class="va">dep</span>,</span>
<span>                                  return_type <span class="op">=</span> <span class="st">"table"</span> ,</span>
<span>                                  contrasts <span class="op">=</span> <span class="st">"W6_vs_PBS"</span></span>
<span>                                  <span class="op">)</span></span>
<span></span>
<span><span class="co">## full table</span></span>
<span><span class="va">full_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_df_wide.html">get_df_wide</a></span><span class="op">(</span><span class="va">dep</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sig_tb</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 54 29</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sig_tb_W6_vs_PBS</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 23 29</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">full_tb</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9595   95</span></span>
<span></span>
<span><span class="co">## Save table in R</span></span>
<span><span class="co"># write.csv(full_tb,"fulldataset.cs",row.names = F)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19043)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                               </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [8] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] org.Mm.eg.db_3.16.0         AnnotationDbi_1.60.0       </span></span>
<span><span class="co">#&gt;  [3] ComplexHeatmap_2.14.0       ggplot2_3.4.1              </span></span>
<span><span class="co">#&gt;  [5] dplyr_1.1.0                 DEP2_0.3.06.12             </span></span>
<span><span class="co">#&gt;  [7] R6_2.5.1                    limma_3.54.1               </span></span>
<span><span class="co">#&gt;  [9] MSnbase_2.24.2              ProtGenerics_1.30.0        </span></span>
<span><span class="co">#&gt; [11] mzR_2.32.0                  Rcpp_1.0.10                </span></span>
<span><span class="co">#&gt; [13] MsCoreUtils_1.10.0          SummarizedExperiment_1.28.0</span></span>
<span><span class="co">#&gt; [15] Biobase_2.58.0              GenomicRanges_1.50.2       </span></span>
<span><span class="co">#&gt; [17] GenomeInfoDb_1.34.9         IRanges_2.32.0             </span></span>
<span><span class="co">#&gt; [19] S4Vectors_0.36.1            BiocGenerics_0.44.0        </span></span>
<span><span class="co">#&gt; [21] MatrixGenerics_1.10.0       matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt; [23] BiocStyle_2.26.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] utf8_1.2.3                  reticulate_1.28            </span></span>
<span><span class="co">#&gt;   [3] gmm_1.7                     tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;   [5] RSQLite_2.3.0               BiocParallel_1.32.5        </span></span>
<span><span class="co">#&gt;   [7] Rtsne_0.16                  norm_1.0-10.0              </span></span>
<span><span class="co">#&gt;   [9] IHW_1.26.0                  munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [11] codetools_0.2-19            ragg_1.2.5                 </span></span>
<span><span class="co">#&gt;  [13] preprocessCore_1.60.2       umap_0.2.10.0              </span></span>
<span><span class="co">#&gt;  [15] withr_2.5.0                 colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [17] highr_0.10                  knitr_1.42                 </span></span>
<span><span class="co">#&gt;  [19] rstudioapi_0.14             mzID_1.36.0                </span></span>
<span><span class="co">#&gt;  [21] labeling_0.4.2              slam_0.1-50                </span></span>
<span><span class="co">#&gt;  [23] bbmle_1.0.25                GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">#&gt;  [25] lpsymphony_1.26.3           mixsqp_0.3-48              </span></span>
<span><span class="co">#&gt;  [27] farver_2.1.1                bit64_4.0.5                </span></span>
<span><span class="co">#&gt;  [29] rprojroot_2.0.3             coda_0.19-4                </span></span>
<span><span class="co">#&gt;  [31] vctrs_0.5.2                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [33] xfun_0.39                   itertools_0.1-3            </span></span>
<span><span class="co">#&gt;  [35] randomForest_4.7-1.1        doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [37] clue_0.3-64                 apeglm_1.20.0              </span></span>
<span><span class="co">#&gt;  [39] invgamma_1.1                locfit_1.5-9.7             </span></span>
<span><span class="co">#&gt;  [41] AnnotationFilter_1.22.0     bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [43] cachem_1.0.6                DelayedArray_0.23.2        </span></span>
<span><span class="co">#&gt;  [45] assertthat_0.2.1            scales_1.2.1               </span></span>
<span><span class="co">#&gt;  [47] gtable_0.3.1                affy_1.76.0                </span></span>
<span><span class="co">#&gt;  [49] sandwich_3.0-2              rlang_1.0.6                </span></span>
<span><span class="co">#&gt;  [51] systemfonts_1.0.4           GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt;  [53] splines_4.2.1               lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [55] impute_1.72.3               BiocManager_1.30.19        </span></span>
<span><span class="co">#&gt;  [57] yaml_2.3.7                  reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [59] tools_4.2.1                 bookdown_0.34              </span></span>
<span><span class="co">#&gt;  [61] affyio_1.68.0               jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [63] RColorBrewer_1.1-3          proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [65] MultiAssayExperiment_1.24.0 plyr_1.8.8                 </span></span>
<span><span class="co">#&gt;  [67] zlibbioc_1.44.0             purrr_1.0.1                </span></span>
<span><span class="co">#&gt;  [69] RCurl_1.98-1.10             openssl_2.0.5              </span></span>
<span><span class="co">#&gt;  [71] GetoptLong_1.0.5            ashr_2.2-54                </span></span>
<span><span class="co">#&gt;  [73] zoo_1.8-11                  ggrepel_0.9.3              </span></span>
<span><span class="co">#&gt;  [75] cluster_2.1.4               fs_1.6.1                   </span></span>
<span><span class="co">#&gt;  [77] magrittr_2.0.3              data.table_1.14.8          </span></span>
<span><span class="co">#&gt;  [79] RSpectra_0.16-1             magick_2.7.4               </span></span>
<span><span class="co">#&gt;  [81] circlize_0.4.15             pcaMethods_1.90.0          </span></span>
<span><span class="co">#&gt;  [83] truncnorm_1.0-8             mvtnorm_1.1-3              </span></span>
<span><span class="co">#&gt;  [85] SQUAREM_2021.1              evaluate_0.20              </span></span>
<span><span class="co">#&gt;  [87] xtable_1.8-4                XML_3.99-0.13              </span></span>
<span><span class="co">#&gt;  [89] emdbook_1.3.12              shape_1.4.6                </span></span>
<span><span class="co">#&gt;  [91] compiler_4.2.1              bdsmatrix_1.3-6            </span></span>
<span><span class="co">#&gt;  [93] tibble_3.1.8                ncdf4_1.21                 </span></span>
<span><span class="co">#&gt;  [95] crayon_1.5.2                htmltools_0.5.4            </span></span>
<span><span class="co">#&gt;  [97] tidyr_1.3.0                 geneplotter_1.76.0         </span></span>
<span><span class="co">#&gt;  [99] DBI_1.1.3                   MASS_7.3-58.2              </span></span>
<span><span class="co">#&gt; [101] tmvtnorm_1.5                Matrix_1.5-3               </span></span>
<span><span class="co">#&gt; [103] cli_3.6.0                   vsn_3.66.0                 </span></span>
<span><span class="co">#&gt; [105] imputeLCMD_2.1              parallel_4.2.1             </span></span>
<span><span class="co">#&gt; [107] igraph_1.4.0                pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [109] pkgdown_2.0.7               GenomicAlignments_1.34.0   </span></span>
<span><span class="co">#&gt; [111] numDeriv_2016.8-1.1         TCseq_1.22.6               </span></span>
<span><span class="co">#&gt; [113] MALDIquant_1.22             foreach_1.5.2              </span></span>
<span><span class="co">#&gt; [115] annotate_1.76.0             bslib_0.4.2                </span></span>
<span><span class="co">#&gt; [117] missForest_1.5              rngtools_1.5.2             </span></span>
<span><span class="co">#&gt; [119] XVector_0.38.0              doRNG_1.8.6                </span></span>
<span><span class="co">#&gt; [121] stringr_1.5.0               digest_0.6.31              </span></span>
<span><span class="co">#&gt; [123] Biostrings_2.66.0           rmarkdown_2.20             </span></span>
<span><span class="co">#&gt; [125] edgeR_3.40.2                Rsamtools_2.14.0           </span></span>
<span><span class="co">#&gt; [127] rjson_0.2.21                lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt; [129] jsonlite_1.8.4              QFeatures_1.8.0            </span></span>
<span><span class="co">#&gt; [131] desc_1.4.2                  askpass_1.1                </span></span>
<span><span class="co">#&gt; [133] fansi_1.0.4                 pillar_1.8.1               </span></span>
<span><span class="co">#&gt; [135] lattice_0.20-45             KEGGREST_1.38.0            </span></span>
<span><span class="co">#&gt; [137] fastmap_1.1.0               httr_1.4.4                 </span></span>
<span><span class="co">#&gt; [139] survival_3.5-3              glue_1.6.2                 </span></span>
<span><span class="co">#&gt; [141] fdrtool_1.2.17              png_0.1-8                  </span></span>
<span><span class="co">#&gt; [143] iterators_1.0.14            glmnet_4.1-6               </span></span>
<span><span class="co">#&gt; [145] bit_4.0.5                   class_7.3-21               </span></span>
<span><span class="co">#&gt; [147] stringi_1.7.12              sass_0.4.5                 </span></span>
<span><span class="co">#&gt; [149] blob_1.2.3                  textshaping_0.3.6          </span></span>
<span><span class="co">#&gt; [151] DESeq2_1.38.3               memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [153] irlba_2.3.5.1               e1071_1.7-13</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Ritchie:2015" class="csl-entry">
Ritchie, M. E., B. Phipson, D. Wu, Y. Hu, C. W. Law, W. Shi, and G. K.
Smyth. 2015. <span>“Limma Powers Differential Expression Analyses for
RNA-Sequencing and Microarray Studies.”</span> Journal Article.
<em>Nucleic Acids Res</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-Zhang:2018" class="csl-entry">
Zhang, X., A. H. Smits, G. B. van Tilburg, H. Ovaa, W. Huber, and M.
Vermeulen. 2018. <span>“Proteome-Wide Identification of Ubiquitin
Interactions Using UbIA-MS.”</span> Journal Article. <em>Nat Protoc</em>
13 (3): 530–50. <a href="https://doi.org/10.1038/nprot.2017.147" class="external-link">https://doi.org/10.1038/nprot.2017.147</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Zhenhuan Feng, Xiaofei Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
