<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ DEP2</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DEP2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.7.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01_Quick_start.html">Quick start</a>
    <a class="dropdown-item" href="../articles/02_Data_import.html">Data import</a>
    <a class="dropdown-item" href="../articles/03_DEP2_analysis.html">DEP2 analysis workflow</a>
    <a class="dropdown-item" href="../articles/04_app_demo.html">DEP2 Application Guide</a>
    <a class="dropdown-item" href="../articles/05_Benchmark_on_spike_in_data.html">A benchmark on spike-in data</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="DEGdata-class.html">DEGdata-class</a></code> 
        </dt>
        <dd>Class "DEGdata"
This class store the test result from DESeq2</dd>
      </dl><dl><dt>
          
          <code><a href="DEP2_add_rejections.html">add_rejections(<i>&lt;SummarizedExperiment&gt;</i>)</a></code> <code><a href="DEP2_add_rejections.html">add_rejections(<i>&lt;DEGdata&gt;</i>)</a></code> 
        </dt>
        <dd>Mark significant proteins</dd>
      </dl><dl><dt>
          
          <code><a href="GS_imp_wrapper.html">GS_imp_wrapper()</a></code> 
        </dt>
        <dd>pre_processing_GS_wrapper</dd>
      </dl><dl><dt>
          
          <code><a href="ID_transform.html">ID_transform()</a></code> 
        </dt>
        <dd>ID transform SE or DEGdata</dd>
      </dl><dl><dt>
          
          <code><a href="NAiszero.html">NAiszero()</a></code> 
        </dt>
        <dd>Transform NA to 0 in assay of a SummarizedExperiment</dd>
      </dl><dl><dt>
          
          <code><a href="Order_cols.html">Order_cols()</a></code> 
        </dt>
        <dd>Set order of experiment design for SummarizedExperiment</dd>
      </dl><dl><dt>
          
          <code><a href="PPInetwork.html">PPInetwork()</a></code> 
        </dt>
        <dd>Draw a network on PPI result</dd>
      </dl><dl><dt>
          
          <code><a href="Silicosis_ExpDesign.html">Silicosis_ExpDesign</a></code> 
        </dt>
        <dd>Silicosis_ExpDesign - experiment design of silicosis mouse proteome</dd>
      </dl><dl><dt>
          
          <code><a href="Silicosis_peptide.html">Silicosis_peptide</a></code> 
        </dt>
        <dd>Silicosis_peptide - peptide quantity data of a silicosis mouse model</dd>
      </dl><dl><dt>
          
          <code><a href="Silicosis_pg.html">Silicosis_pg</a></code> 
        </dt>
        <dd>Silicosis_pg - proteinGroups data of a silicosis mouse model</dd>
      </dl><dl><dt>
          
          <code><a href="Silicosis_phos.html">Silicosis_phos</a></code> 
        </dt>
        <dd>Silicosis_phos - phosphoproteome result of a silicosis mouse model</dd>
      </dl><dl><dt>
          
          <code><a href="aggregate_pe.html">aggregate_pe()</a></code> 
        </dt>
        <dd>Summarize peptide quantity to protein quantity</dd>
      </dl><dl><dt>
          
          <code><a href="annoSpecies_df.html">annoSpecies_df()</a></code> 
        </dt>
        <dd>Species information table</dd>
      </dl><dl><dt>
          
          <code><a href="check_PPI_depends.html">check_PPI_depends()</a></code> 
        </dt>
        <dd>Check required packages for PPI analysis.</dd>
      </dl><dl><dt>
          
          <code><a href="check_RNAseq_depends.html">check_RNAseq_depends()</a></code> 
        </dt>
        <dd>Check required packages for DESeq2.</dd>
      </dl><dl><dt>
          
          <code><a href="check_enrichment_depends.html">check_enrichment_depends()</a></code> 
        </dt>
        <dd>Check required packages for enrichment analysis.</dd>
      </dl><dl><dt>
          
          <code><a href="check_organismDB_depends.html">check_organismDB_depends()</a></code> 
        </dt>
        <dd>Check organism annotation package</dd>
      </dl><dl><dt>
          
          <code><a href="clean_character.html">clean_character()</a></code> 
        </dt>
        <dd>Clean characters in expression columns</dd>
      </dl><dl><dt>
          
          <code><a href="correct_PTM_by_Protein.html">correct_PTM_by_Protein()</a></code> 
        </dt>
        <dd>correct_PTM_by_Protein</dd>
      </dl><dl><dt>
          
          <code><a href="downloadAbsentFile.html">downloadAbsentFile()</a></code> 
        </dt>
        <dd>Title</dd>
      </dl><dl><dt>
          
          <code><a href="filter_pe.html">filter_pe()</a></code> 
        </dt>
        <dd>Filter Qfeatures object on missing values or formula</dd>
      </dl><dl><dt>
          
          <code><a href="filter_se.html">filter_se()</a></code> 
        </dt>
        <dd>Filter SummarizedExperiment on missing values or formula</dd>
      </dl><dl><dt>
          
          <code><a href="get_ORA_result.html">get_ORA_result()</a></code> 
        </dt>
        <dd>Extract significant enrichment terms base on giving threshold</dd>
      </dl><dl><dt>
          
          <code><a href="get_contrast.html">get_contrast()</a></code> 
        </dt>
        <dd>Get contrast(s) from SummarizedExperiment or DEGdata</dd>
      </dl><dl><dt>
          
          <code><a href="get_df_long.html">get_df_long()</a></code> 
        </dt>
        <dd>Generate a long data.frame from a SummarizedExperiment</dd>
      </dl><dl><dt>
          
          <code><a href="get_df_wide.html">get_df_wide()</a></code> 
        </dt>
        <dd>Generate a wide data.frame from a SummarizedExperiment</dd>
      </dl><dl><dt>
          
          <code><a href="get_exdesign_parse.html">get_exdesign_parse()</a></code> 
        </dt>
        <dd>Construct experiment design basic on colnames parse</dd>
      </dl><dl><dt>
          
          <code><a href="get_prefix.html">get_prefix()</a></code> 
        </dt>
        <dd>Obtain the longest common prefix</dd>
      </dl><dl><dt>
          
          <code><a href="get_results.html">get_results()</a></code> 
        </dt>
        <dd>Generate a results table</dd>
      </dl><dl><dt>
          
          <code><a href="get_signicant.html">get_signicant()</a></code> 
        </dt>
        <dd>Extract significant candidates from SummarizedExperiment or DEGdata</dd>
      </dl><dl><dt>
          
          <code><a href="get_suffix.html">get_suffix()</a></code> 
        </dt>
        <dd>Obtain the longest common suffix</dd>
      </dl><dl><dt>
          
          <code><a href="get_tc_cluster.html">get_tc_cluster()</a></code> 
        </dt>
        <dd>Time-course clustering</dd>
      </dl><dl><dt>
          
          <code><a href="impute.html">impute()</a></code> 
        </dt>
        <dd>Impute missing values</dd>
      </dl><dl><dt>
          
          <code><a href="impute_pe.html">impute_pe()</a></code> 
        </dt>
        <dd>Impute a QFeatures object</dd>
      </dl><dl><dt>
          
          <code><a href="load_PPIdata.html">load_PPIdata()</a></code> 
        </dt>
        <dd>Check and load a local STRING data. If local file do not exist, will try to download from <a href="http://stringdb-static.org/download/" class="external-link">STRING</a>.</dd>
      </dl><dl><dt>
          
          <code><a href="make_dds.html">make_dds()</a></code> 
        </dt>
        <dd>Count matrix to DESeqDataSet conversion using an experimental design</dd>
      </dl><dl><dt>
          
          <code><a href="make_dds_parse.html">make_dds_parse()</a></code> 
        </dt>
        <dd>Count matrix to DESeqDataSet conversion by parsing from column names</dd>
      </dl><dl><dt>
          
          <code><a href="make_pe.html">make_pe()</a></code> 
        </dt>
        <dd>Data.frame to QFeatures object conversion using an experimental design</dd>
      </dl><dl><dt>
          
          <code><a href="make_pe_parse.html">make_pe_parse()</a></code> 
        </dt>
        <dd>Data.frame to QFeatures object conversion by parsing column names</dd>
      </dl><dl><dt>
          
          <code><a href="make_se.html">make_se()</a></code> 
        </dt>
        <dd>Data.frame to SummarizedExperiment object
conversion using an experimental design</dd>
      </dl><dl><dt>
          
          <code><a href="make_se_parse.html">make_se_parse()</a></code> 
        </dt>
        <dd>Data.frame to SummarizedExperiment object
conversion using parsing from column names</dd>
      </dl><dl><dt>
          
          <code><a href="make_unique.html">make_unique()</a></code> 
        </dt>
        <dd>Make unique names</dd>
      </dl><dl><dt>
          
          <code><a href="make_unique_ptm.html">make_unique_ptm()</a></code> 
        </dt>
        <dd>Make unique names for a modication-enriched peptide table</dd>
      </dl><dl><dt>
          
          <code><a href="manual_impute.html">manual_impute()</a></code> 
        </dt>
        <dd>Imputation by random draws from a manually defined distribution</dd>
      </dl><dl><dt>
          
          <code><a href="meanSdPlot.html">meanSdPlot()</a></code> 
        </dt>
        <dd>Plot row standard deviations versus row means</dd>
      </dl><dl><dt>
          
          <code><a href="normalize_pe.html">normalize_pe()</a></code> 
        </dt>
        <dd>Normalize a QFeatures object</dd>
      </dl><dl><dt>
          
          <code><a href="normalize_vsn.html">normalize_vsn()</a></code> 
        </dt>
        <dd>Normalization using vsn</dd>
      </dl><dl><dt>
          
          <code><a href="ntf_deg.html">ntf_deg()</a></code> 
        </dt>
        <dd>Assign normalized assay for DEGdata object</dd>
      </dl><dl><dt>
          
          <code><a href="pe2se.html">pe2se()</a></code> 
        </dt>
        <dd>Extract the proteins SummarizedExperiment object from a QFeatures container</dd>
      </dl><dl><dt>
          
          <code><a href="plot_Tsne.html">plot_Tsne()</a></code> 
        </dt>
        <dd>Plot t-Sne</dd>
      </dl><dl><dt>
          
          <code><a href="plot_cond.html">plot_cond()</a></code> 
        </dt>
        <dd>Plot frequency of significant conditions per protein
and the overlap in proteins between conditions</dd>
      </dl><dl><dt>
          
          <code><a href="plot_cond_freq.html">plot_cond_freq()</a></code> 
        </dt>
        <dd>Plot frequency of significant conditions per protein</dd>
      </dl><dl><dt>
          
          <code><a href="plot_cond_overlap.html">plot_cond_overlap()</a></code> 
        </dt>
        <dd>Plot conditions overlap</dd>
      </dl><dl><dt>
          
          <code><a href="plot_cor.html">plot_cor()</a></code> 
        </dt>
        <dd>Plot correlation matrix</dd>
      </dl><dl><dt>
          
          <code><a href="plot_coverage.html">plot_coverage()</a></code> 
        </dt>
        <dd>Plot protein coverage</dd>
      </dl><dl><dt>
          
          <code><a href="plot_cvs.html">plot_cvs()</a></code> 
        </dt>
        <dd>Plot sample coefficient of variation whitin group</dd>
      </dl><dl><dt>
          
          <code><a href="plot_detect.html">plot_detect()</a></code> 
        </dt>
        <dd>Visualize intensities of proteins with missing values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_diff_hist.html">plot_diff_hist()</a></code> 
        </dt>
        <dd>Fit a Gaussian distribution for L2FC of each contrast</dd>
      </dl><dl><dt>
          
          <code><a href="plot_dist.html">plot_dist()</a></code> 
        </dt>
        <dd>Plot Gower's distance matrix</dd>
      </dl><dl><dt>
          
          <code><a href="plot_frequency.html">plot_frequency()</a></code> 
        </dt>
        <dd>Plot protein overlap between samples</dd>
      </dl><dl><dt>
          
          <code><a href="plot_heatmap.html">plot_heatmap(<i>&lt;SummarizedExperiment&gt;</i>)</a></code> <code><a href="plot_heatmap.html">plot_heatmap(<i>&lt;DEGdata&gt;</i>)</a></code> 
        </dt>
        <dd>Plot a heatmap</dd>
      </dl><dl><dt>
          
          <code><a href="plot_imputation.html">plot_imputation()</a></code> 
        </dt>
        <dd>Visualize imputation</dd>
      </dl><dl><dt>
          
          <code><a href="plot_ma_RNA.html">plot_ma_RNA()</a></code> 
        </dt>
        <dd>MA-plot of RNA expression data</dd>
      </dl><dl><dt>
          
          <code><a href="plot_ma_pro.html">plot_ma_pro()</a></code> 
        </dt>
        <dd>MA-plot of quantity data</dd>
      </dl><dl><dt>
          
          <code><a href="plot_missval.html">plot_missval()</a></code> 
        </dt>
        <dd>Plot a heatmap of proteins with missing values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_multi_heatmap.html">plot_multi_heatmap()</a></code> 
        </dt>
        <dd>Plot heatmap of specified genes/proteins across multiple omics results</dd>
      </dl><dl><dt>
          
          <code><a href="plot_multi_venn.html">plot_multi_venn()</a></code> 
        </dt>
        <dd>Plot venn plot of specified genes/proteins across multiple omics results</dd>
      </dl><dl><dt>
          
          <code><a href="plot_norm_distribution.html">plot_norm_distribution()</a></code> 
        </dt>
        <dd>Plot the fit normal for log2 fold change</dd>
      </dl><dl><dt>
          
          <code><a href="plot_normalization.html">plot_normalization()</a></code> 
        </dt>
        <dd>Visualize normalization</dd>
      </dl><dl><dt>
          
          <code><a href="plot_numbers.html">plot_numbers()</a></code> 
        </dt>
        <dd>Plot protein numbers</dd>
      </dl><dl><dt>
          
          <code><a href="plot_pca.html">plot_pca()</a></code> 
        </dt>
        <dd>Plot PCA</dd>
      </dl><dl><dt>
          
          <code><a href="plot_single.html">plot_single()</a></code> 
        </dt>
        <dd>Plot values for a protein/gene of interest</dd>
      </dl><dl><dt>
          
          <code><a href="plot_statistics.html">plot_statistics()</a></code> 
        </dt>
        <dd>Plot the distribution of statistic valuse. develop from statistics_plot</dd>
      </dl><dl><dt>
          
          <code><a href="plot_umap.html">plot_umap()</a></code> 
        </dt>
        <dd>Plot Tsne</dd>
      </dl><dl><dt>
          
          <code><a href="plot_volcano.html">plot_volcano()</a></code> 
        </dt>
        <dd>Volcano plot</dd>
      </dl><dl><dt>
          
          <code><a href="reshape_long2wide.html">reshape_long2wide()</a></code> 
        </dt>
        <dd>Reshape a long table to wide</dd>
      </dl><dl><dt>
          
          <code><a href="rlg_deg.html">rlg_deg()</a></code> 
        </dt>
        <dd>Assign a rlog transformed assay for DEGdata object</dd>
      </dl><dl><dt>
          
          <code><a href="run_app.html">run_app()</a></code> 
        </dt>
        <dd>Run shiny application in DEP2</dd>
      </dl><dl><dt>
          
          <code><a href="test_GSEA.html">test_GSEA()</a></code> 
        </dt>
        <dd>GSEA data</dd>
      </dl><dl><dt>
          
          <code><a href="test_ORA.html">test_ORA()</a></code> 
        </dt>
        <dd>ORA for differenatial test result</dd>
      </dl><dl><dt>
          
          <code><a href="test_PPI.html">test_PPI()</a></code> 
        </dt>
        <dd>Protein-protein interaction analysis</dd>
      </dl><dl><dt>
          
          <code><a href="test_diff.html">test_diff()</a></code> 
        </dt>
        <dd>Differential enrichment/expression test</dd>
      </dl><dl><dt>
          
          <code><a href="test_diff_deg.html">test_diff_deg()</a></code> 
        </dt>
        <dd>Differential expression test on a DESeqDataSet</dd>
      </dl><dl><dt>
          
          <code><a href="theme_DEP1.html">theme_DEP1()</a></code> 
        </dt>
        <dd>DEP ggplot theme 1</dd>
      </dl><dl><dt>
          
          <code><a href="theme_DEP2.html">theme_DEP2()</a></code> 
        </dt>
        <dd>DEP ggplot theme 2</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Zhenhuan Feng, Xiaofei Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

