<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="test_diff performs a differential enrichment/expression test based on
protein/peptide-wise linear models and empirical Bayes
statistics using limma. False Discovery Rates are estimated
using &quot;Strimmer's qvalue&quot;, &quot;Benjamini-Hochberg fdr&quot; or &quot;Storey's qvalue&quot;."><title>Differential enrichment/expression test — test_diff • DEP2</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential enrichment/expression test — test_diff"><meta property="og:description" content="test_diff performs a differential enrichment/expression test based on
protein/peptide-wise linear models and empirical Bayes
statistics using limma. False Discovery Rates are estimated
using &quot;Strimmer's qvalue&quot;, &quot;Benjamini-Hochberg fdr&quot; or &quot;Storey's qvalue&quot;."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DEP2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.7.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/01_Quick_start.html">Quick start</a>
    <a class="dropdown-item" href="../articles/02_Data_import.html">Data import</a>
    <a class="dropdown-item" href="../articles/03_DEP2_analysis.html">DEP2 analysis workflow</a>
    <a class="dropdown-item" href="../articles/04_app_demo.html">DEP2 Application Guide</a>
    <a class="dropdown-item" href="../articles/05_Benchmark_on_spike_in_data.html">A benchmark on spike-in data</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Differential enrichment/expression test</h1>
      
      <div class="d-none name"><code>test_diff.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>test_diff</code> performs a differential enrichment/expression test based on
protein/peptide-wise linear models and empirical Bayes
statistics using <span class="pkg">limma</span>. False Discovery Rates are estimated
using <code>"Strimmer's qvalue"</code>, <code>"Benjamini-Hochberg fdr"</code> or <code>"Storey's qvalue"</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">test_diff</span><span class="op">(</span></span>
<span>  <span class="va">se</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"control"</span>, <span class="st">"manual"</span><span class="op">)</span>,</span>
<span>  control <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  test <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  design_formula <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html" class="external-link">formula</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span>,</span>
<span>  fdr.type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Strimmer's qvalue(t)"</span>, <span class="st">"Strimmer's qvalue(p)"</span>, <span class="st">"BH"</span>, <span class="st">"Storey's qvalue"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>se</dt>
<dd><p>SummarizedExperiment,
Proteomics data from quantity table (output from <code><a href="make_se.html">make_se</a>()</code>,
<code><a href="make_se_parse.html">make_se_parse</a>()</code>), or from re-aggregate peptide quantity from <code>aggregateFeatures()</code> ).
For proteomics data directly from quantity table, it is adviced to first remove
low quanlity and undesired rows by <code><a href="filter_se.html">filter_se</a>()</code>, and missing values are imputed by <code><a href="impute.html">impute</a>()</code>.
Normalization is recommended in both workflow.</p></dd>


<dt>type</dt>
<dd><p>"control", "all" or "manual",
The type of contrasts that will be tested.
This can be all possible pairwise comparisons ("all"),
limited to the comparisons versus the control ("control"), or
manually defined contrasts ("manual").</p></dd>


<dt>control</dt>
<dd><p>Character(1),
The condition to which contrasts are generated if type = "control"
(a control condition would be most appropriate).</p></dd>


<dt>test</dt>
<dd><p>Character,
The contrasts that will be tested if type = "manual".
These should be formatted as "conditionA_vs_conditionB" or
c("conditionA_vs_conditionC", "conditionB_vs_conditionC").</p></dd>


<dt>design_formula</dt>
<dd><p>Formula,
Used to create the design matrix.</p></dd>


<dt>fdr.type</dt>
<dd><p>Character(1), the method to control false discovery rate,
one of "Strimmer's qvalue(t)","Strimmer's qvalue(p)","BH","Storey's qvalue".
"Strimmer's qvalue" calculate fdr vis <code><a href="https://rdrr.io/pkg/fdrtool/man/fdrtool.html" class="external-link">fdrtool</a>()</code> using t-statistic or p values from limma.
"BH" calculate fdr vis <code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a>()</code> using "BH" method .
"Storey's qvalue"  calculate fdr vis <code>stats::qvalue()</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A SummarizedExperiment object or a DEGdata
containing diff, test result from limma and fdr estimates of differential expression.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load example</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_pg</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">Silicosis_pg</span></span></span>
<span class="r-in"><span><span class="va">data_unique</span> <span class="op">&lt;-</span> <span class="fu"><a href="make_unique.html">make_unique</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"Gene.names"</span>, <span class="st">"Protein.IDs"</span>, delim <span class="op">=</span> <span class="st">";"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Make SummarizedExperiment</span></span></span>
<span class="r-in"><span><span class="va">ecols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"LFQ."</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data_unique</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load experiement design</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Silicosis_ExpDesign</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exp_design</span> <span class="op">&lt;-</span> <span class="va">Silicosis_ExpDesign</span></span></span>
<span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="make_se.html">make_se</a></span><span class="op">(</span><span class="va">data_unique</span>, <span class="va">ecols</span>, <span class="va">exp_design</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter and normalize</span></span></span>
<span class="r-in"><span><span class="va">filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="filter_se.html">filter_se</a></span><span class="op">(</span><span class="va">se</span>, thr <span class="op">=</span> <span class="fl">0</span>, fraction <span class="op">=</span> <span class="fl">0.4</span>, filter_formula <span class="op">=</span> <span class="op">~</span> <span class="va">Reverse</span> <span class="op">!=</span> <span class="st">"+"</span> <span class="op">&amp;</span> <span class="va">Potential.contaminant</span><span class="op">!=</span><span class="st">"+"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> filter base on missing number is &lt;= 0 in at least one condition.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> filter base on missing number fraction &lt; 0.4 in each row</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> filter base on giving formula </span>
<span class="r-in"><span><span class="va">norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="normalize_vsn.html">normalize_vsn</a></span><span class="op">(</span><span class="va">filt</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> vsn2: 8762 x 20 matrix (1 stratum). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Please use 'meanSdPlot' to verify the fit.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Impute missing values using different functions</span></span></span>
<span class="r-in"><span><span class="va">imputed</span> <span class="op">&lt;-</span> <span class="fu"><a href="impute.html">impute</a></span><span class="op">(</span><span class="va">norm</span>, fun <span class="op">=</span> <span class="st">"MinProb"</span>, q <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Imputing along margin 2 (samples/columns).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.3026531</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Test for differentially expressed proteins</span></span></span>
<span class="r-in"><span><span class="va">diff</span> <span class="op">&lt;-</span> <span class="fu">test_diff</span><span class="op">(</span><span class="va">imputed</span>, <span class="st">"control"</span>, <span class="st">"PBS"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Tested contrasts: W10_vs_PBS, W2_vs_PBS, W4_vs_PBS, W6_vs_PBS, W9_vs_PBS</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Strimmer's qvalue(t)</span>
<span class="r-in"><span><span class="va">diff</span> <span class="op">&lt;-</span> <span class="fu">test_diff</span><span class="op">(</span><span class="va">imputed</span>, <span class="st">"manual"</span>, test <span class="op">=</span> <span class="st">"PBS_vs_W6"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Tested contrasts: PBS_vs_W6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Strimmer's qvalue(t)</span>
<span class="r-in"><span><span class="va">diff</span> <span class="op">&lt;-</span> <span class="fu">test_diff</span><span class="op">(</span><span class="va">imputed</span>, <span class="st">"manual"</span>, test <span class="op">=</span> <span class="st">"PBS_vs_W6"</span>, fdr.type <span class="op">=</span> <span class="st">"Storey's qvalue"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Tested contrasts: PBS_vs_W6</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Storey's qvalue</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Zhenhuan Feng, Xiaofei Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

