---
title: "DEP2 Shiny Application"
author: "Zhenhuan Feng"
date: "2023-02-01"
package: "DEP2 0.0.2.127"
output: 
  BiocStyle::html_document:
    number_sections: no
    toc: true
    toc_float: TRUE
    self_contained: true
    keep_md: true
    fig_width: 100
    fig_retina: NULL
    
# runtime: shiny
vignette: |
  %\VignetteIndexEntry{DEP2: shiny introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


# Introduction

Welcome to the the shiny app of `DEP2`. Here we provide a interactive platform for the biologists with little programming experience. This shiny application provides differetially analyze pipeline for kinds of mass spectrometry-based quantitative proteomics data, including protein-level quantitative data(proteingroups) and peptide-level quantitative data(modification-specific proteomics). Differetially analyze steps including data filter, normalization, imputation, statistical test and FDR correction. To extend compatibility of inputs generated by various quantitative software, nearly all customization options in `DEP2` are reserved in this application. Default parameters and detail tips are provided to guide unfamiliar users. Generally, the order of the analytical steps are from the top to the bottom in the sidebar. To build an integrative toolkit among, some additional analysis modules based on other packages are also available. Analyze functions are separated into individual modules.

<!-- [简图](简图) -->

# Create analysis modules

Each analysis modules could be directly import by clicking `Add module` and then choose the required analysis module. Modules are divided into two type `Omics analysis` or `Post analysis`, first is focus on differential test, data exploration and visualization, second is the functional analyze and multi-omics comparation. After add a selected module, a new tab will append into the application padge whose name is the combiation of analyze type and serial number. For omics analysis, a unique remark name is required to distinguish results from different omics experiment.

# Omics analysis modules

In this application, four kinds of **omics differential analysis pipeline** are selectable.

-   **Quantificated proteome(based on proteingroup quantification)**

    Differential analysis of protein-level quantitative data, like *intensity*, *LFQ.intensity* or *quantity*.

-   **Quantificated proteome(based on peptide quantification)**

    Differential analysis of proteins from peptide-level quantitative.

-   **Modification-specific proteome(modified peptide quantification)**

    Differential analysis of modification-specific peptides, receiving peptide quantity from modification-specific enrichment.

-   **Additional transcriptome**

    Differential analysis of parallel transcriptome based on package [DEseq2](https://bioconductor.org/packages/DESeq2/).



## Analysis from proteingroups quantitative data.

This modules perform a differential expressed/enriched protein test according protein-level(proteingroups) quantity tabular input(.txt file) generated by various of quantitative proteomics software such as MaxQuant using the generalized approach `limma`. Below is the illustration of graphical user interface of this module.

![Illustration of DEP based on proteingroup](DEP_illu1.png)

Options for DEP2 analyze are in the sidebar area. In the pipeline DEP2 firstly read in the expression/enrichment data, **upload** the `proteingroup.txt`(or other quantitative table) and `ExperimentalDesign.txt`(optional, needless when group experiment by colnames) in the **Files** panel. Few input table requirements is necessary in the application: 1. Tabular quantitative table which columns are separated by tabs(.txt file); 2. Protein names, gene symbol and quantitative values of groups is required in the quantitative table. 

Then assign the columns in the **Columns** panel, like `Name column`, `ID column` and `Expression columns`, 

- `Name column`, the column of **gene symbol**, corresponding the **name** column in the result. If a proteingroup have multiple gene symbol, only the first one will be chosen as the *name*.

- `ID column`, the column of protein ID, like **Uniprot ID**, corresponding the **ID** column in the result. If a proteingroup contains multiple ID, only the first one will be chosen as the *name*.

- `Expression columns`, the expression/enrichment assay, columns of proteingroups 

automatically filled if handling table from MaxQuant. Lastly, select the `imputation`, `FDR correction`, `Thresshold method` and click `Analyze` button. Result table and visual figures locate in the right side of page providing fine-tuning options and download button with detail tips.

## Analysis from peptide quantitative data.

DEP2 also designed another workflow offering proteingroups intensities re-aggregation from peptide quantitative data though the `aggregateFeatures` functions from the [Qfeature](https://bioconductor.org/packages/QFeatures) package. This function aggregate quantitative features from peptide assay into a proteingroup assay though a highly customizable approach. In this workflow, `limma` differential test is performed on the re-aggregated assay without the `proteingroup` table from software. 

Analyze setting in this modules is similar to the previous one, only has an extra step: aggregation. The **Aggregation** panel with options for peptide quantitative/intensity data filter, normalization and aggregation. Expression columns must contain the colnames of quantitative assay. Then click `Aggregate` button to begin aggregation, the computation may take a minute or two. Once aggregation finish, the `Columns` panel will provide the option for control select. 

## Analysis peptide quantitative data.

In DEP2, we expand the differentially enriched test to modification-specific proteomics(like phosphoproteomics or ubiquitomics) directly on modified-peptide intensity. It require the input of enriched modified peptide quantitative table. Pipeline and options just like the protein-level analysis. Modified-peptide intensity assay was extract from uploaded table, filter based on missing value and a localization probability (selectable in the `Cutoff based on` input). The enich peptide intensity could be corrected by the parallel proteome data(a exited proteome analysis result). Imputation and differentially enriched test are performed just like those in protein-level analysis.



## Analysis RNAseq data by DEseq2.

Due to the multiple omics experiment such as perform RNA-sequencing and quantitative proteomics in parallel, we also provide a RNA-sequencing analysis module based on the workflow from `DESeq2` package. Both **gene count** table and **experimental design** table are required. In the first count table, the first column should be **ENSEMBL** id of genes. ID transformation could transform the **ENSEMBL** to **gene symbol**, for the convenience of result interpretation and following post analysis.



# Post analysis modules

Four kinds of **post analysis modules** could be used on omics analysis result.

-   **Over-representation analysis**

    The classical geneontology(GO) enrichment analysis

-   **GSEA**

    The gene set enrichment analysis(GSEA), another way to perform functional enrichment.

-   **Protein-protein interaction**

    Building a PPI network of input genes/proteins based on [string](https://string-db.org/) database.

-   **Timecourse cluster**

    A expression pattern clustering method for timecourse or multiple groups omics experiement.

-   **Integrated analysis**

    A comparison tool for mulit-omics analysis.

Each imported modules could run independently, the results from omics analysis will be transmitted to post analysis modules.

## Over-representation analysis

This module is based on the `enrichGO` functions from the packages [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) and [ReactomePA](https://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html). The GO annotation and gene ID conversion data if from [orgDB](https://www.bioconductor.org/packages/release/BiocViews.html#___OrgDb), the organism annotation database. This module accept both the result genes **symbol** from omics analysis or text input by pasting the candidate genes **symbol**. To speed up the `enrichGO`, we optimized the function to avoid data exctraction from `orgDb` in each run by saving the intermediate data.

The differentially expression/enrichment result in exited omics analysis modules can be transmitted for candidate gene list extraction. We design a drag-drop input based on the dndselectr(https://github.com/serenity-r/dndselectr) package, which provides an intuitive way to set extract options by mouse drag and drop.

![Illustration of ORA input](DEP_illu2_DDN.png)

## GSEA

This module is similar to over-representation analysis module, both based on clusterProfiler package. The difference is that GSEA analysis accept the whole gene list with log2 foldchange but not the candidate symbol.

## Protein-protein interaction

For interactor study, we design a PPI analysis module draw from local string database of different organism, which could be downloaded from string(https://string-db.org/cgi/download), including the gzip files of **protein.links.detailed**, **protein.aliases**, **protein.info**. Such files must store in the PPIdata folder under the path of DEP2. 

## Timecourse cluster

This module is a kind of expression expression pattern clustering for timecourse study or multiple-groups experiment. The expression/enrichment assay is clustered by the `timeclust` function in **TCseq** package.

## Integrated analysis

Multi-omics study is the trend in biological pathway research and drug target discovery. DEP2 give a easy-used module to compare and visualization result candidate from different omics in the shiny.












